// @operandinc/sdk/dist/esm/index/v1/index_pb.js
// @generated by protoc-gen-es v0.1.1 with parameter "target=ts+js"
// @generated from file index/v1/index.proto (package index.v1, syntax proto3)
/* eslint-disable */
/* @ts-nocheck */
import { Message, proto3, Timestamp } from "@bufbuild/protobuf";
/**
 * @generated from enum index.v1.ObjectType
 */
export var ObjectType;
(function (ObjectType) {
    /**
     * @generated from enum value: OBJECT_TYPE_UNSPECIFIED = 0;
     */
    ObjectType[ObjectType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * @generated from enum value: OBJECT_TYPE_COLLECTION = 1;
     */
    ObjectType[ObjectType["COLLECTION"] = 1] = "COLLECTION";
    /**
     * @generated from enum value: OBJECT_TYPE_TEXT = 2;
     */
    ObjectType[ObjectType["TEXT"] = 2] = "TEXT";
    /**
     * @generated from enum value: OBJECT_TYPE_HTML = 3;
     */
    ObjectType[ObjectType["HTML"] = 3] = "HTML";
    /**
     * @generated from enum value: OBJECT_TYPE_IMAGE = 4;
     */
    ObjectType[ObjectType["IMAGE"] = 4] = "IMAGE";
    /**
     * @generated from enum value: OBJECT_TYPE_PDF = 5;
     */
    ObjectType[ObjectType["PDF"] = 5] = "PDF";
    /**
     * @generated from enum value: OBJECT_TYPE_AUDIO = 6;
     */
    ObjectType[ObjectType["AUDIO"] = 6] = "AUDIO";
    /**
     * Slack is a unique object type, which requires some additional work
     * on the client side to get working. First off, when the object is first
     * created, we'll do a fetch of all the messages that we have access to,
     * i.e. all the channels that we've been added to. Following this, we await
     * webhook events from Slack, sent via the Webhook API method of this service.
     * We expect the client to essentially reverse proxy certain slack webhooks to
     * the correct object ID, which we'll then use to update the object. Specifically,
     * we support webhooks when we're added to a new channel, and when a new message
     * is posted to a channel we're part of.
     *
     * @generated from enum value: OBJECT_TYPE_SLACK = 7;
     */
    ObjectType[ObjectType["SLACK"] = 7] = "SLACK";
})(ObjectType || (ObjectType = {}));
// Retrieve enum metadata with: proto3.getEnumType(ObjectType)
proto3.util.setEnumType(ObjectType, "index.v1.ObjectType", [
    { no: 0, name: "OBJECT_TYPE_UNSPECIFIED" },
    { no: 1, name: "OBJECT_TYPE_COLLECTION" },
    { no: 2, name: "OBJECT_TYPE_TEXT" },
    { no: 3, name: "OBJECT_TYPE_HTML" },
    { no: 4, name: "OBJECT_TYPE_IMAGE" },
    { no: 5, name: "OBJECT_TYPE_PDF" },
    { no: 6, name: "OBJECT_TYPE_AUDIO" },
    { no: 7, name: "OBJECT_TYPE_SLACK" },
]);
/**
 * @generated from enum index.v1.ObjectStatus
 */
export var ObjectStatus;
(function (ObjectStatus) {
    /**
     * @generated from enum value: OBJECT_STATUS_UNSPECIFIED = 0;
     */
    ObjectStatus[ObjectStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * @generated from enum value: OBJECT_STATUS_QUEUED = 1;
     */
    ObjectStatus[ObjectStatus["QUEUED"] = 1] = "QUEUED";
    /**
     * @generated from enum value: OBJECT_STATUS_INDEXING = 2;
     */
    ObjectStatus[ObjectStatus["INDEXING"] = 2] = "INDEXING";
    /**
     * @generated from enum value: OBJECT_STATUS_READY = 3;
     */
    ObjectStatus[ObjectStatus["READY"] = 3] = "READY";
    /**
     * @generated from enum value: OBJECT_STATUS_ERROR = 4;
     */
    ObjectStatus[ObjectStatus["ERROR"] = 4] = "ERROR";
})(ObjectStatus || (ObjectStatus = {}));
// Retrieve enum metadata with: proto3.getEnumType(ObjectStatus)
proto3.util.setEnumType(ObjectStatus, "index.v1.ObjectStatus", [
    { no: 0, name: "OBJECT_STATUS_UNSPECIFIED" },
    { no: 1, name: "OBJECT_STATUS_QUEUED" },
    { no: 2, name: "OBJECT_STATUS_INDEXING" },
    { no: 3, name: "OBJECT_STATUS_READY" },
    { no: 4, name: "OBJECT_STATUS_ERROR" },
]);
/**
 * @generated from message index.v1.UpsertObjectRequest
 */
export class UpsertObjectRequest extends Message {
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new UpsertObjectRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new UpsertObjectRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new UpsertObjectRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(UpsertObjectRequest, a, b);
    }
}
UpsertObjectRequest.runtime = proto3;
UpsertObjectRequest.typeName = "index.v1.UpsertObjectRequest";
UpsertObjectRequest.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "existing_id", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 2, name: "parent_id", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 3, name: "object_type", kind: "enum", T: proto3.getEnumType(ObjectType), opt: true },
    { no: 4, name: "metadata", kind: "message", T: ObjectMetadata, opt: true },
    { no: 5, name: "properties", kind: "message", T: Properties, opt: true },
    { no: 6, name: "label", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 7, name: "unique_label", kind: "scalar", T: 8 /* ScalarType.BOOL */, opt: true },
]);
/**
 * @generated from message index.v1.UpsertObjectResponse
 */
export class UpsertObjectResponse extends Message {
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new UpsertObjectResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new UpsertObjectResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new UpsertObjectResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(UpsertObjectResponse, a, b);
    }
}
UpsertObjectResponse.runtime = proto3;
UpsertObjectResponse.typeName = "index.v1.UpsertObjectResponse";
UpsertObjectResponse.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "object", kind: "message", T: Object$ },
]);
/**
 * @generated from message index.v1.BulkUpsertObjectsRequest
 */
export class BulkUpsertObjectsRequest extends Message {
    constructor(data) {
        super();
        /**
         * TODO(Morgan): Add sync option (i.e. wait and return responses).
         *
         * @generated from field: repeated index.v1.UpsertObjectRequest requests = 1;
         */
        this.requests = [];
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new BulkUpsertObjectsRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new BulkUpsertObjectsRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new BulkUpsertObjectsRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(BulkUpsertObjectsRequest, a, b);
    }
}
BulkUpsertObjectsRequest.runtime = proto3;
BulkUpsertObjectsRequest.typeName = "index.v1.BulkUpsertObjectsRequest";
BulkUpsertObjectsRequest.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "requests", kind: "message", T: UpsertObjectRequest, repeated: true },
]);
/**
 * Currently, no way of checking if this operation was successful.
 *
 * @generated from message index.v1.BulkUpsertObjectsResponse
 */
export class BulkUpsertObjectsResponse extends Message {
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new BulkUpsertObjectsResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new BulkUpsertObjectsResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new BulkUpsertObjectsResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(BulkUpsertObjectsResponse, a, b);
    }
}
BulkUpsertObjectsResponse.runtime = proto3;
BulkUpsertObjectsResponse.typeName = "index.v1.BulkUpsertObjectsResponse";
BulkUpsertObjectsResponse.fields = proto3.util.newFieldList(() => []);
/**
 * @generated from message index.v1.ListObjectsRequest
 */
export class ListObjectsRequest extends Message {
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new ListObjectsRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ListObjectsRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ListObjectsRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ListObjectsRequest, a, b);
    }
}
ListObjectsRequest.runtime = proto3;
ListObjectsRequest.typeName = "index.v1.ListObjectsRequest";
ListObjectsRequest.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "parent_id", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 2, name: "cursor", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 3, name: "limit", kind: "scalar", T: 5 /* ScalarType.INT32 */, opt: true },
    { no: 4, name: "label", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
]);
/**
 * @generated from message index.v1.ListObjectsResponse
 */
export class ListObjectsResponse extends Message {
    constructor(data) {
        super();
        /**
         * @generated from field: repeated index.v1.Object objects = 1;
         */
        this.objects = [];
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new ListObjectsResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ListObjectsResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ListObjectsResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ListObjectsResponse, a, b);
    }
}
ListObjectsResponse.runtime = proto3;
ListObjectsResponse.typeName = "index.v1.ListObjectsResponse";
ListObjectsResponse.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "objects", kind: "message", T: Object$, repeated: true },
    { no: 2, name: "next_cursor", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
]);
/**
 * @generated from message index.v1.GetObjectsRequest
 */
export class GetObjectsRequest extends Message {
    constructor(data) {
        super();
        /**
         * @generated from field: repeated string object_ids = 1;
         */
        this.objectIds = [];
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new GetObjectsRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new GetObjectsRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new GetObjectsRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(GetObjectsRequest, a, b);
    }
}
GetObjectsRequest.runtime = proto3;
GetObjectsRequest.typeName = "index.v1.GetObjectsRequest";
GetObjectsRequest.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "object_ids", kind: "scalar", T: 9 /* ScalarType.STRING */, repeated: true },
]);
/**
 * @generated from message index.v1.GetObjectsResponse
 */
export class GetObjectsResponse extends Message {
    constructor(data) {
        super();
        /**
         * @generated from field: map<string, index.v1.Object> objects = 1;
         */
        this.objects = {};
        /**
         * @generated from field: repeated string missing = 2;
         */
        this.missing = [];
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new GetObjectsResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new GetObjectsResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new GetObjectsResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(GetObjectsResponse, a, b);
    }
}
GetObjectsResponse.runtime = proto3;
GetObjectsResponse.typeName = "index.v1.GetObjectsResponse";
GetObjectsResponse.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "objects", kind: "map", K: 9 /* ScalarType.STRING */, V: { kind: "message", T: Object$ } },
    { no: 2, name: "missing", kind: "scalar", T: 9 /* ScalarType.STRING */, repeated: true },
]);
/**
 * @generated from message index.v1.DeleteObjectRequest
 */
export class DeleteObjectRequest extends Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string id = 1;
         */
        this.id = "";
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new DeleteObjectRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new DeleteObjectRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new DeleteObjectRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(DeleteObjectRequest, a, b);
    }
}
DeleteObjectRequest.runtime = proto3;
DeleteObjectRequest.typeName = "index.v1.DeleteObjectRequest";
DeleteObjectRequest.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
]);
/**
 * @generated from message index.v1.DeleteObjectResponse
 */
export class DeleteObjectResponse extends Message {
    constructor(data) {
        super();
        /**
         * @generated from field: int32 deleted = 1;
         */
        this.deleted = 0;
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new DeleteObjectResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new DeleteObjectResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new DeleteObjectResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(DeleteObjectResponse, a, b);
    }
}
DeleteObjectResponse.runtime = proto3;
DeleteObjectResponse.typeName = "index.v1.DeleteObjectResponse";
DeleteObjectResponse.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "deleted", kind: "scalar", T: 5 /* ScalarType.INT32 */ },
]);
/**
 * @generated from message index.v1.BulkDeleteObjectsRequest
 */
export class BulkDeleteObjectsRequest extends Message {
    constructor(data) {
        super();
        /**
         * Label can be an exact match, or use '%' as a wildcard.
         * For example, 'foo%' will match all labels that start with 'foo'.
         * This is case sensitive.
         *
         * @generated from field: string label = 2;
         */
        this.label = "";
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new BulkDeleteObjectsRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new BulkDeleteObjectsRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new BulkDeleteObjectsRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(BulkDeleteObjectsRequest, a, b);
    }
}
BulkDeleteObjectsRequest.runtime = proto3;
BulkDeleteObjectsRequest.typeName = "index.v1.BulkDeleteObjectsRequest";
BulkDeleteObjectsRequest.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "parent_id", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 2, name: "label", kind: "scalar", T: 9 /* ScalarType.STRING */ },
]);
/**
 * @generated from message index.v1.BulkDeleteObjectsResponse
 */
export class BulkDeleteObjectsResponse extends Message {
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new BulkDeleteObjectsResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new BulkDeleteObjectsResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new BulkDeleteObjectsResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(BulkDeleteObjectsResponse, a, b);
    }
}
BulkDeleteObjectsResponse.runtime = proto3;
BulkDeleteObjectsResponse.typeName = "index.v1.BulkDeleteObjectsResponse";
BulkDeleteObjectsResponse.fields = proto3.util.newFieldList(() => []);
/**
 * @generated from message index.v1.CountObjectsRequest
 */
export class CountObjectsRequest extends Message {
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new CountObjectsRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new CountObjectsRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new CountObjectsRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(CountObjectsRequest, a, b);
    }
}
CountObjectsRequest.runtime = proto3;
CountObjectsRequest.typeName = "index.v1.CountObjectsRequest";
CountObjectsRequest.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "parent_id", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 2, name: "label", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
]);
/**
 * @generated from message index.v1.CountObjectsResponse
 */
export class CountObjectsResponse extends Message {
    constructor(data) {
        super();
        /**
         * @generated from field: int32 count = 1;
         */
        this.count = 0;
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new CountObjectsResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new CountObjectsResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new CountObjectsResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(CountObjectsResponse, a, b);
    }
}
CountObjectsResponse.runtime = proto3;
CountObjectsResponse.typeName = "index.v1.CountObjectsResponse";
CountObjectsResponse.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "count", kind: "scalar", T: 5 /* ScalarType.INT32 */ },
]);
/**
 * @generated from message index.v1.WebhookRequest
 */
export class WebhookRequest extends Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string object_id = 1;
         */
        this.objectId = "";
        /**
         * @generated from field: string operation = 2;
         */
        this.operation = "";
        /**
         * @generated from field: bytes body = 3;
         */
        this.body = new Uint8Array(0);
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new WebhookRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new WebhookRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new WebhookRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(WebhookRequest, a, b);
    }
}
WebhookRequest.runtime = proto3;
WebhookRequest.typeName = "index.v1.WebhookRequest";
WebhookRequest.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "object_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "operation", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 3, name: "body", kind: "scalar", T: 12 /* ScalarType.BYTES */ },
]);
/**
 * @generated from message index.v1.WebhookResponse
 */
export class WebhookResponse extends Message {
    constructor(data) {
        super();
        /**
         * @generated from field: bytes body = 1;
         */
        this.body = new Uint8Array(0);
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new WebhookResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new WebhookResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new WebhookResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(WebhookResponse, a, b);
    }
}
WebhookResponse.runtime = proto3;
WebhookResponse.typeName = "index.v1.WebhookResponse";
WebhookResponse.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "body", kind: "scalar", T: 12 /* ScalarType.BYTES */ },
]);
/**
 * @generated from message index.v1.SearchRequest
 */
export class SearchRequest extends Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string query = 1;
         */
        this.query = "";
        /**
         * @generated from field: repeated string roots = 2;
         */
        this.roots = [];
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new SearchRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SearchRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SearchRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(SearchRequest, a, b);
    }
}
SearchRequest.runtime = proto3;
SearchRequest.typeName = "index.v1.SearchRequest";
SearchRequest.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "query", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "roots", kind: "scalar", T: 9 /* ScalarType.STRING */, repeated: true },
    { no: 3, name: "session_id", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 4, name: "filter", kind: "message", T: Filter, opt: true },
    { no: 5, name: "limit", kind: "scalar", T: 5 /* ScalarType.INT32 */, opt: true },
]);
/**
 * @generated from message index.v1.SearchResponse
 */
export class SearchResponse extends Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string search_id = 1;
         */
        this.searchId = "";
        /**
         * @generated from field: repeated index.v1.SearchResponse.Match matches = 2;
         */
        this.matches = [];
        /**
         * @generated from field: map<string, index.v1.Object> objects = 3;
         */
        this.objects = {};
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new SearchResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SearchResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SearchResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(SearchResponse, a, b);
    }
}
SearchResponse.runtime = proto3;
SearchResponse.typeName = "index.v1.SearchResponse";
SearchResponse.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "search_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "matches", kind: "message", T: SearchResponse_Match, repeated: true },
    { no: 3, name: "objects", kind: "map", K: 9 /* ScalarType.STRING */, V: { kind: "message", T: Object$ } },
]);
/**
 * @generated from message index.v1.SearchResponse.Match
 */
export class SearchResponse_Match extends Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string match_id = 1;
         */
        this.matchId = "";
        /**
         * @generated from field: string object_id = 2;
         */
        this.objectId = "";
        /**
         * @generated from field: string content = 3;
         */
        this.content = "";
        /**
         * @generated from field: float score = 4;
         */
        this.score = 0;
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new SearchResponse_Match().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SearchResponse_Match().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SearchResponse_Match().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(SearchResponse_Match, a, b);
    }
}
SearchResponse_Match.runtime = proto3;
SearchResponse_Match.typeName = "index.v1.SearchResponse.Match";
SearchResponse_Match.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "match_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "object_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 3, name: "content", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 4, name: "score", kind: "scalar", T: 2 /* ScalarType.FLOAT */ },
    { no: 5, name: "extra", kind: "message", T: Properties, opt: true },
]);
/**
 * @generated from message index.v1.AnswerRequest
 */
export class AnswerRequest extends Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string query = 1;
         */
        this.query = "";
        /**
         * @generated from field: repeated string roots = 2;
         */
        this.roots = [];
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new AnswerRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new AnswerRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new AnswerRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(AnswerRequest, a, b);
    }
}
AnswerRequest.runtime = proto3;
AnswerRequest.typeName = "index.v1.AnswerRequest";
AnswerRequest.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "query", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "roots", kind: "scalar", T: 9 /* ScalarType.STRING */, repeated: true },
    { no: 3, name: "session_id", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 4, name: "filter", kind: "message", T: Filter, opt: true },
]);
/**
 * @generated from message index.v1.AnswerResponse
 */
export class AnswerResponse extends Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string answer_id = 1;
         */
        this.answerId = "";
        /**
         * @generated from field: string answer = 2;
         */
        this.answer = "";
        /**
         * @generated from field: float confidence = 3;
         */
        this.confidence = 0;
        /**
         * @generated from field: repeated index.v1.Object sources = 4;
         */
        this.sources = [];
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new AnswerResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new AnswerResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new AnswerResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(AnswerResponse, a, b);
    }
}
AnswerResponse.runtime = proto3;
AnswerResponse.typeName = "index.v1.AnswerResponse";
AnswerResponse.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "answer_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "answer", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 3, name: "confidence", kind: "scalar", T: 2 /* ScalarType.FLOAT */ },
    { no: 4, name: "sources", kind: "message", T: Object$, repeated: true },
]);
/**
 * @generated from message index.v1.SearchFeedback
 */
export class SearchFeedback extends Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string search_id = 1;
         */
        this.searchId = "";
        /**
         * @generated from field: string clicked_match_id = 2;
         */
        this.clickedMatchId = "";
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new SearchFeedback().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SearchFeedback().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SearchFeedback().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(SearchFeedback, a, b);
    }
}
SearchFeedback.runtime = proto3;
SearchFeedback.typeName = "index.v1.SearchFeedback";
SearchFeedback.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "search_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "clicked_match_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
]);
/**
 * @generated from message index.v1.AnswerFeedback
 */
export class AnswerFeedback extends Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string answer_id = 1;
         */
        this.answerId = "";
        /**
         * @generated from field: bool positive_reaction = 2;
         */
        this.positiveReaction = false;
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new AnswerFeedback().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new AnswerFeedback().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new AnswerFeedback().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(AnswerFeedback, a, b);
    }
}
AnswerFeedback.runtime = proto3;
AnswerFeedback.typeName = "index.v1.AnswerFeedback";
AnswerFeedback.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "answer_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "positive_reaction", kind: "scalar", T: 8 /* ScalarType.BOOL */ },
]);
/**
 * @generated from message index.v1.FeedbackRequest
 */
export class FeedbackRequest extends Message {
    constructor(data) {
        super();
        /**
         * @generated from oneof index.v1.FeedbackRequest.feedback
         */
        this.feedback = { case: undefined };
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new FeedbackRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new FeedbackRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new FeedbackRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(FeedbackRequest, a, b);
    }
}
FeedbackRequest.runtime = proto3;
FeedbackRequest.typeName = "index.v1.FeedbackRequest";
FeedbackRequest.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "search", kind: "message", T: SearchFeedback, oneof: "feedback" },
    { no: 2, name: "answer", kind: "message", T: AnswerFeedback, oneof: "feedback" },
]);
/**
 * @generated from message index.v1.FeedbackResponse
 */
export class FeedbackResponse extends Message {
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new FeedbackResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new FeedbackResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new FeedbackResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(FeedbackResponse, a, b);
    }
}
FeedbackResponse.runtime = proto3;
FeedbackResponse.typeName = "index.v1.FeedbackResponse";
FeedbackResponse.fields = proto3.util.newFieldList(() => []);
/**
 * @generated from message index.v1.ObjectMetadata
 */
export class ObjectMetadata extends Message {
    constructor(data) {
        super();
        /**
         * @generated from oneof index.v1.ObjectMetadata.metadata
         */
        this.metadata = { case: undefined };
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new ObjectMetadata().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ObjectMetadata().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ObjectMetadata().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ObjectMetadata, a, b);
    }
}
ObjectMetadata.runtime = proto3;
ObjectMetadata.typeName = "index.v1.ObjectMetadata";
ObjectMetadata.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "collection", kind: "message", T: CollectionObjectMetadata, oneof: "metadata" },
    { no: 2, name: "text", kind: "message", T: TextObjectMetadata, oneof: "metadata" },
    { no: 3, name: "html", kind: "message", T: HTMLObjectMetadata, oneof: "metadata" },
    { no: 4, name: "image", kind: "message", T: ImageObjectMetadata, oneof: "metadata" },
    { no: 5, name: "pdf", kind: "message", T: PDFObjectMetadata, oneof: "metadata" },
    { no: 6, name: "audio", kind: "message", T: AudioObjectMetadata, oneof: "metadata" },
    { no: 7, name: "slack", kind: "message", T: SlackObjectMetadata, oneof: "metadata" },
]);
/**
 * @generated from message index.v1.CollectionObjectMetadata
 */
export class CollectionObjectMetadata extends Message {
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new CollectionObjectMetadata().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new CollectionObjectMetadata().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new CollectionObjectMetadata().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(CollectionObjectMetadata, a, b);
    }
}
CollectionObjectMetadata.runtime = proto3;
CollectionObjectMetadata.typeName = "index.v1.CollectionObjectMetadata";
CollectionObjectMetadata.fields = proto3.util.newFieldList(() => []);
/**
 * @generated from message index.v1.TextObjectMetadata
 */
export class TextObjectMetadata extends Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string text = 1;
         */
        this.text = "";
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new TextObjectMetadata().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new TextObjectMetadata().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new TextObjectMetadata().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(TextObjectMetadata, a, b);
    }
}
TextObjectMetadata.runtime = proto3;
TextObjectMetadata.typeName = "index.v1.TextObjectMetadata";
TextObjectMetadata.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "text", kind: "scalar", T: 9 /* ScalarType.STRING */ },
]);
/**
 * @generated from message index.v1.HTMLObjectMetadata
 */
export class HTMLObjectMetadata extends Message {
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new HTMLObjectMetadata().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new HTMLObjectMetadata().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new HTMLObjectMetadata().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(HTMLObjectMetadata, a, b);
    }
}
HTMLObjectMetadata.runtime = proto3;
HTMLObjectMetadata.typeName = "index.v1.HTMLObjectMetadata";
HTMLObjectMetadata.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "html", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 2, name: "url", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
]);
/**
 * @generated from message index.v1.ImageObjectMetadata
 */
export class ImageObjectMetadata extends Message {
    constructor(data) {
        super();
        /**
         * Supports: JPEG/JPG, PNG.
         *
         * @generated from field: string url = 1;
         */
        this.url = "";
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new ImageObjectMetadata().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ImageObjectMetadata().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ImageObjectMetadata().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ImageObjectMetadata, a, b);
    }
}
ImageObjectMetadata.runtime = proto3;
ImageObjectMetadata.typeName = "index.v1.ImageObjectMetadata";
ImageObjectMetadata.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "url", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "alt", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
]);
/**
 * @generated from message index.v1.PDFObjectMetadata
 */
export class PDFObjectMetadata extends Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string url = 1;
         */
        this.url = "";
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new PDFObjectMetadata().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new PDFObjectMetadata().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new PDFObjectMetadata().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(PDFObjectMetadata, a, b);
    }
}
PDFObjectMetadata.runtime = proto3;
PDFObjectMetadata.typeName = "index.v1.PDFObjectMetadata";
PDFObjectMetadata.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "url", kind: "scalar", T: 9 /* ScalarType.STRING */ },
]);
/**
 * @generated from message index.v1.AudioObjectMetadata
 */
export class AudioObjectMetadata extends Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string url = 1;
         */
        this.url = "";
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new AudioObjectMetadata().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new AudioObjectMetadata().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new AudioObjectMetadata().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(AudioObjectMetadata, a, b);
    }
}
AudioObjectMetadata.runtime = proto3;
AudioObjectMetadata.typeName = "index.v1.AudioObjectMetadata";
AudioObjectMetadata.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "url", kind: "scalar", T: 9 /* ScalarType.STRING */ },
]);
/**
 * @generated from message index.v1.SlackObjectMetadata
 */
export class SlackObjectMetadata extends Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string bot_token = 1;
         */
        this.botToken = "";
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new SlackObjectMetadata().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SlackObjectMetadata().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SlackObjectMetadata().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(SlackObjectMetadata, a, b);
    }
}
SlackObjectMetadata.runtime = proto3;
SlackObjectMetadata.typeName = "index.v1.SlackObjectMetadata";
SlackObjectMetadata.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "bot_token", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "bot_user_id", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
]);
/**
 * @generated from message index.v1.Properties
 */
export class Properties extends Message {
    constructor(data) {
        super();
        /**
         * @generated from field: map<string, index.v1.Property> properties = 1;
         */
        this.properties = {};
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new Properties().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new Properties().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new Properties().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(Properties, a, b);
    }
}
Properties.runtime = proto3;
Properties.typeName = "index.v1.Properties";
Properties.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "properties", kind: "map", K: 9 /* ScalarType.STRING */, V: { kind: "message", T: Property } },
]);
/**
 * @generated from message index.v1.TextArray
 */
export class TextArray extends Message {
    constructor(data) {
        super();
        /**
         * @generated from field: repeated string values = 1;
         */
        this.values = [];
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new TextArray().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new TextArray().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new TextArray().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(TextArray, a, b);
    }
}
TextArray.runtime = proto3;
TextArray.typeName = "index.v1.TextArray";
TextArray.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "values", kind: "scalar", T: 9 /* ScalarType.STRING */, repeated: true },
]);
/**
 * @generated from message index.v1.NumberArray
 */
export class NumberArray extends Message {
    constructor(data) {
        super();
        /**
         * @generated from field: repeated double values = 1;
         */
        this.values = [];
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new NumberArray().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new NumberArray().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new NumberArray().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(NumberArray, a, b);
    }
}
NumberArray.runtime = proto3;
NumberArray.typeName = "index.v1.NumberArray";
NumberArray.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "values", kind: "scalar", T: 1 /* ScalarType.DOUBLE */, repeated: true },
]);
/**
 * @generated from message index.v1.Property
 */
export class Property extends Message {
    constructor(data) {
        super();
        /**
         * @generated from oneof index.v1.Property.value
         */
        this.value = { case: undefined };
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new Property().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new Property().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new Property().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(Property, a, b);
    }
}
Property.runtime = proto3;
Property.typeName = "index.v1.Property";
Property.fields = proto3.util.newFieldList(() => [
    { no: 2, name: "text", kind: "scalar", T: 9 /* ScalarType.STRING */, oneof: "value" },
    { no: 3, name: "number", kind: "scalar", T: 1 /* ScalarType.DOUBLE */, oneof: "value" },
    { no: 4, name: "text_array", kind: "message", T: TextArray, oneof: "value" },
    { no: 5, name: "number_array", kind: "message", T: NumberArray, oneof: "value" },
]);
/**
 * @generated from message index.v1.KeyedProperty
 */
export class KeyedProperty extends Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string key = 1;
         */
        this.key = "";
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new KeyedProperty().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new KeyedProperty().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new KeyedProperty().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(KeyedProperty, a, b);
    }
}
KeyedProperty.runtime = proto3;
KeyedProperty.typeName = "index.v1.KeyedProperty";
KeyedProperty.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "key", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "property", kind: "message", T: Property },
]);
/**
 * @generated from message index.v1.Object
 */
export class Object$ extends Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string id = 1;
         */
        this.id = "";
        /**
         * @generated from field: index.v1.ObjectType type = 5;
         */
        this.type = ObjectType.UNSPECIFIED;
        /**
         * @generated from field: index.v1.ObjectStatus status = 8;
         */
        this.status = ObjectStatus.UNSPECIFIED;
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new Object$().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new Object$().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new Object$().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(Object$, a, b);
    }
}
Object$.runtime = proto3;
Object$.typeName = "index.v1.Object";
Object$.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "created_at", kind: "message", T: Timestamp },
    { no: 3, name: "updated_at", kind: "message", T: Timestamp, opt: true },
    { no: 4, name: "parent_id", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 5, name: "type", kind: "enum", T: proto3.getEnumType(ObjectType) },
    { no: 6, name: "metadata", kind: "message", T: ObjectMetadata },
    { no: 7, name: "properties", kind: "message", T: Properties },
    { no: 8, name: "status", kind: "enum", T: proto3.getEnumType(ObjectStatus) },
    { no: 9, name: "label", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
]);
/**
 * @generated from message index.v1.Filter
 */
export class Filter extends Message {
    constructor(data) {
        super();
        /**
         * @generated from field: repeated index.v1.Condition conditions = 1;
         */
        this.conditions = [];
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new Filter().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new Filter().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new Filter().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(Filter, a, b);
    }
}
Filter.runtime = proto3;
Filter.typeName = "index.v1.Filter";
Filter.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "conditions", kind: "message", T: Condition, repeated: true },
]);
/**
 * @generated from message index.v1.Range
 */
export class Range extends Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string key = 1;
         */
        this.key = "";
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new Range().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new Range().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new Range().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(Range, a, b);
    }
}
Range.runtime = proto3;
Range.typeName = "index.v1.Range";
Range.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "key", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "lt", kind: "scalar", T: 1 /* ScalarType.DOUBLE */, opt: true },
    { no: 3, name: "lte", kind: "scalar", T: 1 /* ScalarType.DOUBLE */, opt: true },
    { no: 4, name: "gt", kind: "scalar", T: 1 /* ScalarType.DOUBLE */, opt: true },
    { no: 5, name: "gte", kind: "scalar", T: 1 /* ScalarType.DOUBLE */, opt: true },
]);
/**
 * @generated from message index.v1.Condition
 */
export class Condition extends Message {
    constructor(data) {
        super();
        /**
         * @generated from oneof index.v1.Condition.condition
         */
        this.condition = { case: undefined };
        proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new Condition().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new Condition().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new Condition().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(Condition, a, b);
    }
}
Condition.runtime = proto3;
Condition.typeName = "index.v1.Condition";
Condition.fields = proto3.util.newFieldList(() => [
    { no: 1, name: "property", kind: "message", T: KeyedProperty, oneof: "condition" },
    { no: 2, name: "range", kind: "message", T: Range, oneof: "condition" },
    { no: 3, name: "and", kind: "message", T: Filter, oneof: "condition" },
    { no: 4, name: "or", kind: "message", T: Filter, oneof: "condition" },
    { no: 5, name: "not", kind: "message", T: Filter, oneof: "condition" },
]);
