// @operandinc/sdk/dist/esm/operand/v1/index_connectweb.js
// @generated by protoc-gen-connect-web v0.2.1 with parameter "target=ts"
// @generated from file operand/v1/index.proto (package operand.v1, syntax proto3)
/* eslint-disable */
/* @ts-nocheck */
import { AvailableExplorationsRequest, AvailableExplorationsResponse, CreateExplorationRequest, CreateExplorationResponse, CreateIndexRequest, CreateIndexResponse, DeleteExplorationRequest, DeleteExplorationResponse, DeleteIndexRequest, DeleteIndexResponse, GetExplorationRequest, GetExplorationResponse, GetIndexRequest, GetIndexResponse, InviteRequest, InviteResponse, ListExplorationsRequest, ListExplorationsResponse, ListIndexesRequest, ListIndexesResponse, SubscribeRequest, SubscribeResponse, SubscribersOfRequest, SubscribersOfResponse, SubscriptionsRequest, SubscriptionsResponse, UnsubscribeRequest, UnsubscribeResponse, UpdateIndexRequest, UpdateIndexResponse } from "./index_pb.js";
import { MethodKind } from "@bufbuild/protobuf";
/**
 * IndexService manages the users set of configured indexes. This includes
 * both the indexes they've personally created, as well as the indexes they've
 * subscribed to (i.e. public indexes).
 *
 * @generated from service operand.v1.IndexService
 */
export const IndexService = {
    typeName: "operand.v1.IndexService",
    methods: {
        /**
         * Allows the user to create a new index that they own.
         *
         * @generated from rpc operand.v1.IndexService.CreateIndex
         */
        createIndex: {
            name: "CreateIndex",
            I: CreateIndexRequest,
            O: CreateIndexResponse,
            kind: MethodKind.Unary,
        },
        /**
         * ListIndexes returns a list of indexes that this or another user owns.
         *
         * @generated from rpc operand.v1.IndexService.ListIndexes
         */
        listIndexes: {
            name: "ListIndexes",
            I: ListIndexesRequest,
            O: ListIndexesResponse,
            kind: MethodKind.Unary,
        },
        /**
         * GetIndex returns the details of a single index. The response also includes
         * additional information about the index, i.e. usage information, social data,
         * if requested by the caller.
         *
         * @generated from rpc operand.v1.IndexService.GetIndex
         */
        getIndex: {
            name: "GetIndex",
            I: GetIndexRequest,
            O: GetIndexResponse,
            kind: MethodKind.Unary,
        },
        /**
         * UpdateIndex allows the user to update the details of an index. For now,
         * this operation is very limited, and only allows the user to toggle a few
         * smaller metadata-like settings on the index.
         *
         * @generated from rpc operand.v1.IndexService.UpdateIndex
         */
        updateIndex: {
            name: "UpdateIndex",
            I: UpdateIndexRequest,
            O: UpdateIndexResponse,
            kind: MethodKind.Unary,
        },
        /**
         * DeleteIndex allows the user to delete an index. This operation will complete
         * instantly, but it will take a few minutes for the index to be fully deleted.
         * Another note is that any subscribers to this index will be notified that the
         * index has been deleted, and will therefore be unable to use it any more.
         *
         * @generated from rpc operand.v1.IndexService.DeleteIndex
         */
        deleteIndex: {
            name: "DeleteIndex",
            I: DeleteIndexRequest,
            O: DeleteIndexResponse,
            kind: MethodKind.Unary,
        },
        /**
         * Subscriptions returns the list of indexes that this, or another user has subscribed to.
         *
         * @generated from rpc operand.v1.IndexService.Subscriptions
         */
        subscriptions: {
            name: "Subscriptions",
            I: SubscriptionsRequest,
            O: SubscriptionsResponse,
            kind: MethodKind.Unary,
        },
        /**
         * Subscribe allows the user to subscribe to a public index.
         *
         * @generated from rpc operand.v1.IndexService.Subscribe
         */
        subscribe: {
            name: "Subscribe",
            I: SubscribeRequest,
            O: SubscribeResponse,
            kind: MethodKind.Unary,
        },
        /**
         * Unsubscribe allows the user to unsubscribe from a public index.
         * This endpoint also allows the owner of an index to manually unsubscribe a subscriber,
         * essentially, a "kick" operation. This is important for private indexes (i.e. teams).
         *
         * @generated from rpc operand.v1.IndexService.Unsubscribe
         */
        unsubscribe: {
            name: "Unsubscribe",
            I: UnsubscribeRequest,
            O: UnsubscribeResponse,
            kind: MethodKind.Unary,
        },
        /**
         * Invite allows a user to invite another user to subscribe to an index.
         * A few notes:
         * - The user must be the owner of the index.
         * - The index need not be public, it's possible to invite users to private indexes.
         * - The user must not already be subscribed to the index (an error will be returned).
         * - The target user need not exist, an invitation will be sent to them, and they will
         *   be given the option to create an account before accepting the invitation.
         *
         * @generated from rpc operand.v1.IndexService.Invite
         */
        invite: {
            name: "Invite",
            I: InviteRequest,
            O: InviteResponse,
            kind: MethodKind.Unary,
        },
        /**
         * SubscribersOf returns the list of users that have subscribed to an index.
         *
         * @generated from rpc operand.v1.IndexService.SubscribersOf
         */
        subscribersOf: {
            name: "SubscribersOf",
            I: SubscribersOfRequest,
            O: SubscribersOfResponse,
            kind: MethodKind.Unary,
        },
        /**
         * AvailableExplorations returns the list of explorations that are available for an index.
         *
         * @generated from rpc operand.v1.IndexService.AvailableExplorations
         */
        availableExplorations: {
            name: "AvailableExplorations",
            I: AvailableExplorationsRequest,
            O: AvailableExplorationsResponse,
            kind: MethodKind.Unary,
        },
        /**
         * CreateExploration allows the user to create a new exploration for an index.
         *
         * @generated from rpc operand.v1.IndexService.CreateExploration
         */
        createExploration: {
            name: "CreateExploration",
            I: CreateExplorationRequest,
            O: CreateExplorationResponse,
            kind: MethodKind.Unary,
        },
        /**
         * DeleteExploration allows the user to delete an exploration from an index.
         *
         * @generated from rpc operand.v1.IndexService.DeleteExploration
         */
        deleteExploration: {
            name: "DeleteExploration",
            I: DeleteExplorationRequest,
            O: DeleteExplorationResponse,
            kind: MethodKind.Unary,
        },
        /**
         * ListExplorations returns the list of explorations that have been created for an index.
         *
         * @generated from rpc operand.v1.IndexService.ListExplorations
         */
        listExplorations: {
            name: "ListExplorations",
            I: ListExplorationsRequest,
            O: ListExplorationsResponse,
            kind: MethodKind.Unary,
        },
        /**
         * GetExploration returns the details of a single exploration that was created for an index.
         *
         * @generated from rpc operand.v1.IndexService.GetExploration
         */
        getExploration: {
            name: "GetExploration",
            I: GetExplorationRequest,
            O: GetExplorationResponse,
            kind: MethodKind.Unary,
        },
    }
};
