// @operandinc/sdk/dist/cjs/operand/v1/index_pb.js
"use strict";
// @generated by protoc-gen-es v0.1.1 with parameter "target=ts+js"
// @generated from file operand/v1/index.proto (package operand.v1, syntax proto3)
/* eslint-disable */
/* @ts-nocheck */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetExplorationResponse = exports.GetExplorationRequest = exports.ListExplorationsResponse = exports.ListExplorationsRequest = exports.DeleteExplorationResponse = exports.DeleteExplorationRequest = exports.CreateExplorationResponse = exports.CreateExplorationRequest = exports.ExplorationOptions = exports.Exploration = exports.ConceptsExplorationData_Concept = exports.ConceptsExplorationData_Summary = exports.ConceptsExplorationData_Snippet = exports.ConceptsExplorationData = exports.ExplorationData = exports.ConceptsExplorationParameters_SummaryKind = exports.ConceptsExplorationParameters = exports.ExplorationParameters = exports.AvailableExplorationsResponse = exports.AvailableExplorationsRequest = exports.SubscribersOfResponse = exports.SubscribersOfRequest = exports.InviteResponse = exports.InviteRequest = exports.UnsubscribeResponse = exports.UnsubscribeRequest = exports.SubscribeResponse = exports.SubscribeRequest = exports.SubscriptionsResponse = exports.SubscriptionsRequest = exports.DeleteIndexResponse = exports.DeleteIndexRequest = exports.UpdateIndexResponse = exports.UpdateIndexRequest = exports.GetIndexResponse = exports.GetIndexRequest = exports.ListIndexesResponse = exports.ListIndexesRequest = exports.CreateIndexResponse = exports.Index = exports.IndexStats = exports.Subscription = exports.CreateIndexRequest = exports.IndexOptions = exports.ExplorationStatus = exports.ExplorationKind = void 0;
const protobuf_1 = require("@bufbuild/protobuf");
const object_pb_js_1 = require("./object_pb.js");
/**
 * @generated from enum operand.v1.ExplorationKind
 */
var ExplorationKind;
(function (ExplorationKind) {
    /**
     * @generated from enum value: EXPLORATION_KIND_UNSPECIFIED = 0;
     */
    ExplorationKind[ExplorationKind["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * @generated from enum value: EXPLORATION_KIND_CONCEPTS = 1;
     */
    ExplorationKind[ExplorationKind["CONCEPTS"] = 1] = "CONCEPTS";
})(ExplorationKind = exports.ExplorationKind || (exports.ExplorationKind = {}));
// Retrieve enum metadata with: proto3.getEnumType(ExplorationKind)
protobuf_1.proto3.util.setEnumType(ExplorationKind, "operand.v1.ExplorationKind", [
    { no: 0, name: "EXPLORATION_KIND_UNSPECIFIED" },
    { no: 1, name: "EXPLORATION_KIND_CONCEPTS" },
]);
/**
 * @generated from enum operand.v1.ExplorationStatus
 */
var ExplorationStatus;
(function (ExplorationStatus) {
    /**
     * @generated from enum value: EXPLORATION_STATUS_UNSPECIFIED = 0;
     */
    ExplorationStatus[ExplorationStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * @generated from enum value: EXPLORATION_STATUS_PENDING = 1;
     */
    ExplorationStatus[ExplorationStatus["PENDING"] = 1] = "PENDING";
    /**
     * @generated from enum value: EXPLORATION_STATUS_RUNNING = 2;
     */
    ExplorationStatus[ExplorationStatus["RUNNING"] = 2] = "RUNNING";
    /**
     * @generated from enum value: EXPLORATION_STATUS_COMPLETED = 3;
     */
    ExplorationStatus[ExplorationStatus["COMPLETED"] = 3] = "COMPLETED";
    /**
     * @generated from enum value: EXPLORATION_STATUS_FAILED = 4;
     */
    ExplorationStatus[ExplorationStatus["FAILED"] = 4] = "FAILED";
})(ExplorationStatus = exports.ExplorationStatus || (exports.ExplorationStatus = {}));
// Retrieve enum metadata with: proto3.getEnumType(ExplorationStatus)
protobuf_1.proto3.util.setEnumType(ExplorationStatus, "operand.v1.ExplorationStatus", [
    { no: 0, name: "EXPLORATION_STATUS_UNSPECIFIED" },
    { no: 1, name: "EXPLORATION_STATUS_PENDING" },
    { no: 2, name: "EXPLORATION_STATUS_RUNNING" },
    { no: 3, name: "EXPLORATION_STATUS_COMPLETED" },
    { no: 4, name: "EXPLORATION_STATUS_FAILED" },
]);
/**
 * @generated from message operand.v1.IndexOptions
 */
class IndexOptions extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new IndexOptions().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new IndexOptions().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new IndexOptions().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(IndexOptions, a, b);
    }
}
exports.IndexOptions = IndexOptions;
IndexOptions.runtime = protobuf_1.proto3;
IndexOptions.typeName = "operand.v1.IndexOptions";
IndexOptions.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "include_owner", kind: "scalar", T: 8 /* ScalarType.BOOL */, opt: true },
    { no: 2, name: "include_subscription", kind: "scalar", T: 8 /* ScalarType.BOOL */, opt: true },
    { no: 3, name: "include_stats", kind: "scalar", T: 8 /* ScalarType.BOOL */, opt: true },
]);
/**
 * @generated from message operand.v1.CreateIndexRequest
 */
class CreateIndexRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string name = 1;
         */
        this.name = "";
        /**
         * @generated from field: string description = 2;
         */
        this.description = "";
        /**
         * @generated from field: bool public = 3;
         */
        this.public = false;
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new CreateIndexRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new CreateIndexRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new CreateIndexRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(CreateIndexRequest, a, b);
    }
}
exports.CreateIndexRequest = CreateIndexRequest;
CreateIndexRequest.runtime = protobuf_1.proto3;
CreateIndexRequest.typeName = "operand.v1.CreateIndexRequest";
CreateIndexRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "name", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "description", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 3, name: "public", kind: "scalar", T: 8 /* ScalarType.BOOL */ },
    { no: 4, name: "options", kind: "message", T: IndexOptions, opt: true },
    { no: 5, name: "image_url", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
]);
/**
 * @generated from message operand.v1.Subscription
 */
class Subscription extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: bool notifications = 2;
         */
        this.notifications = false;
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new Subscription().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new Subscription().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new Subscription().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(Subscription, a, b);
    }
}
exports.Subscription = Subscription;
Subscription.runtime = protobuf_1.proto3;
Subscription.typeName = "operand.v1.Subscription";
Subscription.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "created_at", kind: "message", T: protobuf_1.Timestamp },
    { no: 2, name: "notifications", kind: "scalar", T: 8 /* ScalarType.BOOL */ },
]);
/**
 * @generated from message operand.v1.IndexStats
 */
class IndexStats extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: int32 subscribers = 1;
         */
        this.subscribers = 0;
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new IndexStats().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new IndexStats().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new IndexStats().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(IndexStats, a, b);
    }
}
exports.IndexStats = IndexStats;
IndexStats.runtime = protobuf_1.proto3;
IndexStats.typeName = "operand.v1.IndexStats";
IndexStats.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "subscribers", kind: "scalar", T: 5 /* ScalarType.INT32 */ },
]);
/**
 * @generated from message operand.v1.Index
 */
class Index extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string public_id = 1;
         */
        this.publicId = "";
        /**
         * @generated from field: bool public = 4;
         */
        this.public = false;
        /**
         * @generated from field: string name = 5;
         */
        this.name = "";
        /**
         * @generated from field: string description = 6;
         */
        this.description = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new Index().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new Index().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new Index().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(Index, a, b);
    }
}
exports.Index = Index;
Index.runtime = protobuf_1.proto3;
Index.typeName = "operand.v1.Index";
Index.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "public_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "owner", kind: "message", T: object_pb_js_1.UserProfile, opt: true },
    { no: 3, name: "created_at", kind: "message", T: protobuf_1.Timestamp },
    { no: 4, name: "public", kind: "scalar", T: 8 /* ScalarType.BOOL */ },
    { no: 5, name: "name", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 6, name: "description", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 7, name: "subscription", kind: "message", T: Subscription, opt: true },
    { no: 8, name: "stats", kind: "message", T: IndexStats, opt: true },
    { no: 9, name: "image_url", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
]);
/**
 * @generated from message operand.v1.CreateIndexResponse
 */
class CreateIndexResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new CreateIndexResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new CreateIndexResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new CreateIndexResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(CreateIndexResponse, a, b);
    }
}
exports.CreateIndexResponse = CreateIndexResponse;
CreateIndexResponse.runtime = protobuf_1.proto3;
CreateIndexResponse.typeName = "operand.v1.CreateIndexResponse";
CreateIndexResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "index", kind: "message", T: Index },
]);
/**
 * @generated from message operand.v1.ListIndexesRequest
 */
class ListIndexesRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new ListIndexesRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ListIndexesRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ListIndexesRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(ListIndexesRequest, a, b);
    }
}
exports.ListIndexesRequest = ListIndexesRequest;
ListIndexesRequest.runtime = protobuf_1.proto3;
ListIndexesRequest.typeName = "operand.v1.ListIndexesRequest";
ListIndexesRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "owned_by_user", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 2, name: "options", kind: "message", T: IndexOptions, opt: true },
]);
/**
 * @generated from message operand.v1.ListIndexesResponse
 */
class ListIndexesResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: repeated operand.v1.Index indexes = 1;
         */
        this.indexes = [];
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new ListIndexesResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ListIndexesResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ListIndexesResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(ListIndexesResponse, a, b);
    }
}
exports.ListIndexesResponse = ListIndexesResponse;
ListIndexesResponse.runtime = protobuf_1.proto3;
ListIndexesResponse.typeName = "operand.v1.ListIndexesResponse";
ListIndexesResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "indexes", kind: "message", T: Index, repeated: true },
]);
/**
 * @generated from message operand.v1.GetIndexRequest
 */
class GetIndexRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new GetIndexRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new GetIndexRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new GetIndexRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(GetIndexRequest, a, b);
    }
}
exports.GetIndexRequest = GetIndexRequest;
GetIndexRequest.runtime = protobuf_1.proto3;
GetIndexRequest.typeName = "operand.v1.GetIndexRequest";
GetIndexRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "public_id", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 2, name: "options", kind: "message", T: IndexOptions, opt: true },
    { no: 3, name: "invite_code", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
]);
/**
 * @generated from message operand.v1.GetIndexResponse
 */
class GetIndexResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new GetIndexResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new GetIndexResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new GetIndexResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(GetIndexResponse, a, b);
    }
}
exports.GetIndexResponse = GetIndexResponse;
GetIndexResponse.runtime = protobuf_1.proto3;
GetIndexResponse.typeName = "operand.v1.GetIndexResponse";
GetIndexResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "index", kind: "message", T: Index },
]);
/**
 * @generated from message operand.v1.UpdateIndexRequest
 */
class UpdateIndexRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string public_id = 1;
         */
        this.publicId = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new UpdateIndexRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new UpdateIndexRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new UpdateIndexRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(UpdateIndexRequest, a, b);
    }
}
exports.UpdateIndexRequest = UpdateIndexRequest;
UpdateIndexRequest.runtime = protobuf_1.proto3;
UpdateIndexRequest.typeName = "operand.v1.UpdateIndexRequest";
UpdateIndexRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "public_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "name", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 3, name: "description", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 4, name: "public", kind: "scalar", T: 8 /* ScalarType.BOOL */, opt: true },
    { no: 5, name: "options", kind: "message", T: IndexOptions, opt: true },
    { no: 6, name: "image_url", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
]);
/**
 * @generated from message operand.v1.UpdateIndexResponse
 */
class UpdateIndexResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new UpdateIndexResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new UpdateIndexResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new UpdateIndexResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(UpdateIndexResponse, a, b);
    }
}
exports.UpdateIndexResponse = UpdateIndexResponse;
UpdateIndexResponse.runtime = protobuf_1.proto3;
UpdateIndexResponse.typeName = "operand.v1.UpdateIndexResponse";
UpdateIndexResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "index", kind: "message", T: Index },
]);
/**
 * @generated from message operand.v1.DeleteIndexRequest
 */
class DeleteIndexRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string public_id = 1;
         */
        this.publicId = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new DeleteIndexRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new DeleteIndexRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new DeleteIndexRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(DeleteIndexRequest, a, b);
    }
}
exports.DeleteIndexRequest = DeleteIndexRequest;
DeleteIndexRequest.runtime = protobuf_1.proto3;
DeleteIndexRequest.typeName = "operand.v1.DeleteIndexRequest";
DeleteIndexRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "public_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
]);
/**
 * @generated from message operand.v1.DeleteIndexResponse
 */
class DeleteIndexResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new DeleteIndexResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new DeleteIndexResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new DeleteIndexResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(DeleteIndexResponse, a, b);
    }
}
exports.DeleteIndexResponse = DeleteIndexResponse;
DeleteIndexResponse.runtime = protobuf_1.proto3;
DeleteIndexResponse.typeName = "operand.v1.DeleteIndexResponse";
DeleteIndexResponse.fields = protobuf_1.proto3.util.newFieldList(() => []);
/**
 * @generated from message operand.v1.SubscriptionsRequest
 */
class SubscriptionsRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new SubscriptionsRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SubscriptionsRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SubscriptionsRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(SubscriptionsRequest, a, b);
    }
}
exports.SubscriptionsRequest = SubscriptionsRequest;
SubscriptionsRequest.runtime = protobuf_1.proto3;
SubscriptionsRequest.typeName = "operand.v1.SubscriptionsRequest";
SubscriptionsRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "user_id", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 2, name: "options", kind: "message", T: IndexOptions, opt: true },
]);
/**
 * @generated from message operand.v1.SubscriptionsResponse
 */
class SubscriptionsResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: repeated operand.v1.Index indexes = 1;
         */
        this.indexes = [];
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new SubscriptionsResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SubscriptionsResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SubscriptionsResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(SubscriptionsResponse, a, b);
    }
}
exports.SubscriptionsResponse = SubscriptionsResponse;
SubscriptionsResponse.runtime = protobuf_1.proto3;
SubscriptionsResponse.typeName = "operand.v1.SubscriptionsResponse";
SubscriptionsResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "indexes", kind: "message", T: Index, repeated: true },
]);
/**
 * @generated from message operand.v1.SubscribeRequest
 */
class SubscribeRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new SubscribeRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SubscribeRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SubscribeRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(SubscribeRequest, a, b);
    }
}
exports.SubscribeRequest = SubscribeRequest;
SubscribeRequest.runtime = protobuf_1.proto3;
SubscribeRequest.typeName = "operand.v1.SubscribeRequest";
SubscribeRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "public_id", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 2, name: "notifications", kind: "scalar", T: 8 /* ScalarType.BOOL */, opt: true },
    { no: 3, name: "invite_token", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
]);
/**
 * @generated from message operand.v1.SubscribeResponse
 */
class SubscribeResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new SubscribeResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SubscribeResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SubscribeResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(SubscribeResponse, a, b);
    }
}
exports.SubscribeResponse = SubscribeResponse;
SubscribeResponse.runtime = protobuf_1.proto3;
SubscribeResponse.typeName = "operand.v1.SubscribeResponse";
SubscribeResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "subscription", kind: "message", T: Subscription },
]);
/**
 * @generated from message operand.v1.UnsubscribeRequest
 */
class UnsubscribeRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string public_id = 1;
         */
        this.publicId = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new UnsubscribeRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new UnsubscribeRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new UnsubscribeRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(UnsubscribeRequest, a, b);
    }
}
exports.UnsubscribeRequest = UnsubscribeRequest;
UnsubscribeRequest.runtime = protobuf_1.proto3;
UnsubscribeRequest.typeName = "operand.v1.UnsubscribeRequest";
UnsubscribeRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "public_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "target_user_id", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
]);
/**
 * @generated from message operand.v1.UnsubscribeResponse
 */
class UnsubscribeResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new UnsubscribeResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new UnsubscribeResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new UnsubscribeResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(UnsubscribeResponse, a, b);
    }
}
exports.UnsubscribeResponse = UnsubscribeResponse;
UnsubscribeResponse.runtime = protobuf_1.proto3;
UnsubscribeResponse.typeName = "operand.v1.UnsubscribeResponse";
UnsubscribeResponse.fields = protobuf_1.proto3.util.newFieldList(() => []);
/**
 * @generated from message operand.v1.InviteRequest
 */
class InviteRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string public_id = 1;
         */
        this.publicId = "";
        /**
         * @generated from oneof operand.v1.InviteRequest.target
         */
        this.target = { case: undefined };
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new InviteRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new InviteRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new InviteRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(InviteRequest, a, b);
    }
}
exports.InviteRequest = InviteRequest;
InviteRequest.runtime = protobuf_1.proto3;
InviteRequest.typeName = "operand.v1.InviteRequest";
InviteRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "public_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "user_public_id", kind: "scalar", T: 9 /* ScalarType.STRING */, oneof: "target" },
    { no: 3, name: "email_address", kind: "scalar", T: 9 /* ScalarType.STRING */, oneof: "target" },
]);
/**
 * @generated from message operand.v1.InviteResponse
 */
class InviteResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new InviteResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new InviteResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new InviteResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(InviteResponse, a, b);
    }
}
exports.InviteResponse = InviteResponse;
InviteResponse.runtime = protobuf_1.proto3;
InviteResponse.typeName = "operand.v1.InviteResponse";
InviteResponse.fields = protobuf_1.proto3.util.newFieldList(() => []);
/**
 * @generated from message operand.v1.SubscribersOfRequest
 */
class SubscribersOfRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string public_id = 1;
         */
        this.publicId = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new SubscribersOfRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SubscribersOfRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SubscribersOfRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(SubscribersOfRequest, a, b);
    }
}
exports.SubscribersOfRequest = SubscribersOfRequest;
SubscribersOfRequest.runtime = protobuf_1.proto3;
SubscribersOfRequest.typeName = "operand.v1.SubscribersOfRequest";
SubscribersOfRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "public_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
]);
/**
 * @generated from message operand.v1.SubscribersOfResponse
 */
class SubscribersOfResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: repeated operand.v1.UserProfile subscribers = 1;
         */
        this.subscribers = [];
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new SubscribersOfResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SubscribersOfResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SubscribersOfResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(SubscribersOfResponse, a, b);
    }
}
exports.SubscribersOfResponse = SubscribersOfResponse;
SubscribersOfResponse.runtime = protobuf_1.proto3;
SubscribersOfResponse.typeName = "operand.v1.SubscribersOfResponse";
SubscribersOfResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "subscribers", kind: "message", T: object_pb_js_1.UserProfile, repeated: true },
]);
/**
 * @generated from message operand.v1.AvailableExplorationsRequest
 */
class AvailableExplorationsRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string index_id = 1;
         */
        this.indexId = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new AvailableExplorationsRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new AvailableExplorationsRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new AvailableExplorationsRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(AvailableExplorationsRequest, a, b);
    }
}
exports.AvailableExplorationsRequest = AvailableExplorationsRequest;
AvailableExplorationsRequest.runtime = protobuf_1.proto3;
AvailableExplorationsRequest.typeName = "operand.v1.AvailableExplorationsRequest";
AvailableExplorationsRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "index_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
]);
/**
 * @generated from message operand.v1.AvailableExplorationsResponse
 */
class AvailableExplorationsResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: repeated operand.v1.ExplorationKind kinds = 1;
         */
        this.kinds = [];
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new AvailableExplorationsResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new AvailableExplorationsResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new AvailableExplorationsResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(AvailableExplorationsResponse, a, b);
    }
}
exports.AvailableExplorationsResponse = AvailableExplorationsResponse;
AvailableExplorationsResponse.runtime = protobuf_1.proto3;
AvailableExplorationsResponse.typeName = "operand.v1.AvailableExplorationsResponse";
AvailableExplorationsResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "kinds", kind: "enum", T: protobuf_1.proto3.getEnumType(ExplorationKind), repeated: true },
]);
/**
 * @generated from message operand.v1.ExplorationParameters
 */
class ExplorationParameters extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from oneof operand.v1.ExplorationParameters.kind
         */
        this.kind = { case: undefined };
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new ExplorationParameters().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ExplorationParameters().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ExplorationParameters().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(ExplorationParameters, a, b);
    }
}
exports.ExplorationParameters = ExplorationParameters;
ExplorationParameters.runtime = protobuf_1.proto3;
ExplorationParameters.typeName = "operand.v1.ExplorationParameters";
ExplorationParameters.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "concepts", kind: "message", T: ConceptsExplorationParameters, oneof: "kind" },
]);
/**
 * @generated from message operand.v1.ConceptsExplorationParameters
 */
class ConceptsExplorationParameters extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: int32 target_size = 1;
         */
        this.targetSize = 0;
        /**
         * @generated from field: repeated operand.v1.ConceptsExplorationParameters.SummaryKind summary_kinds = 2;
         */
        this.summaryKinds = [];
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new ConceptsExplorationParameters().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ConceptsExplorationParameters().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ConceptsExplorationParameters().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(ConceptsExplorationParameters, a, b);
    }
}
exports.ConceptsExplorationParameters = ConceptsExplorationParameters;
ConceptsExplorationParameters.runtime = protobuf_1.proto3;
ConceptsExplorationParameters.typeName = "operand.v1.ConceptsExplorationParameters";
ConceptsExplorationParameters.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "target_size", kind: "scalar", T: 5 /* ScalarType.INT32 */ },
    { no: 2, name: "summary_kinds", kind: "enum", T: protobuf_1.proto3.getEnumType(ConceptsExplorationParameters_SummaryKind), repeated: true },
    { no: 3, name: "filter", kind: "message", T: object_pb_js_1.Filter, opt: true },
]);
/**
 * @generated from enum operand.v1.ConceptsExplorationParameters.SummaryKind
 */
var ConceptsExplorationParameters_SummaryKind;
(function (ConceptsExplorationParameters_SummaryKind) {
    /**
     * @generated from enum value: SUMMARY_KIND_UNSPECIFIED = 0;
     */
    ConceptsExplorationParameters_SummaryKind[ConceptsExplorationParameters_SummaryKind["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * @generated from enum value: SUMMARY_KIND_TOPIC_STRING = 1;
     */
    ConceptsExplorationParameters_SummaryKind[ConceptsExplorationParameters_SummaryKind["TOPIC_STRING"] = 1] = "TOPIC_STRING";
    /**
     * @generated from enum value: SUMMARY_KIND_MARKDOWN = 2;
     */
    ConceptsExplorationParameters_SummaryKind[ConceptsExplorationParameters_SummaryKind["MARKDOWN"] = 2] = "MARKDOWN";
})(ConceptsExplorationParameters_SummaryKind = exports.ConceptsExplorationParameters_SummaryKind || (exports.ConceptsExplorationParameters_SummaryKind = {}));
// Retrieve enum metadata with: proto3.getEnumType(ConceptsExplorationParameters_SummaryKind)
protobuf_1.proto3.util.setEnumType(ConceptsExplorationParameters_SummaryKind, "operand.v1.ConceptsExplorationParameters.SummaryKind", [
    { no: 0, name: "SUMMARY_KIND_UNSPECIFIED" },
    { no: 1, name: "SUMMARY_KIND_TOPIC_STRING" },
    { no: 2, name: "SUMMARY_KIND_MARKDOWN" },
]);
/**
 * @generated from message operand.v1.ExplorationData
 */
class ExplorationData extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from oneof operand.v1.ExplorationData.kind
         */
        this.kind = { case: undefined };
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new ExplorationData().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ExplorationData().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ExplorationData().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(ExplorationData, a, b);
    }
}
exports.ExplorationData = ExplorationData;
ExplorationData.runtime = protobuf_1.proto3;
ExplorationData.typeName = "operand.v1.ExplorationData";
ExplorationData.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "concepts", kind: "message", T: ConceptsExplorationData, oneof: "kind" },
]);
/**
 * @generated from message operand.v1.ConceptsExplorationData
 */
class ConceptsExplorationData extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: repeated operand.v1.ConceptsExplorationData.Concept concepts = 1;
         */
        this.concepts = [];
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new ConceptsExplorationData().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ConceptsExplorationData().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ConceptsExplorationData().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(ConceptsExplorationData, a, b);
    }
}
exports.ConceptsExplorationData = ConceptsExplorationData;
ConceptsExplorationData.runtime = protobuf_1.proto3;
ConceptsExplorationData.typeName = "operand.v1.ConceptsExplorationData";
ConceptsExplorationData.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "concepts", kind: "message", T: ConceptsExplorationData_Concept, repeated: true },
]);
/**
 * @generated from message operand.v1.ConceptsExplorationData.Snippet
 */
class ConceptsExplorationData_Snippet extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string text = 1;
         */
        this.text = "";
        /**
         * @generated from field: string object_id = 2;
         */
        this.objectId = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new ConceptsExplorationData_Snippet().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ConceptsExplorationData_Snippet().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ConceptsExplorationData_Snippet().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(ConceptsExplorationData_Snippet, a, b);
    }
}
exports.ConceptsExplorationData_Snippet = ConceptsExplorationData_Snippet;
ConceptsExplorationData_Snippet.runtime = protobuf_1.proto3;
ConceptsExplorationData_Snippet.typeName = "operand.v1.ConceptsExplorationData.Snippet";
ConceptsExplorationData_Snippet.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "text", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "object_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
]);
/**
 * @generated from message operand.v1.ConceptsExplorationData.Summary
 */
class ConceptsExplorationData_Summary extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from oneof operand.v1.ConceptsExplorationData.Summary.kind
         */
        this.kind = { case: undefined };
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new ConceptsExplorationData_Summary().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ConceptsExplorationData_Summary().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ConceptsExplorationData_Summary().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(ConceptsExplorationData_Summary, a, b);
    }
}
exports.ConceptsExplorationData_Summary = ConceptsExplorationData_Summary;
ConceptsExplorationData_Summary.runtime = protobuf_1.proto3;
ConceptsExplorationData_Summary.typeName = "operand.v1.ConceptsExplorationData.Summary";
ConceptsExplorationData_Summary.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "topic_string", kind: "scalar", T: 9 /* ScalarType.STRING */, oneof: "kind" },
    { no: 2, name: "markdown", kind: "scalar", T: 9 /* ScalarType.STRING */, oneof: "kind" },
]);
/**
 * @generated from message operand.v1.ConceptsExplorationData.Concept
 */
class ConceptsExplorationData_Concept extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: repeated operand.v1.ConceptsExplorationData.Snippet snippets = 1;
         */
        this.snippets = [];
        /**
         * @generated from field: map<string, operand.v1.Object> objects = 2;
         */
        this.objects = {};
        /**
         * @generated from field: repeated operand.v1.ConceptsExplorationData.Summary summaries = 3;
         */
        this.summaries = [];
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new ConceptsExplorationData_Concept().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ConceptsExplorationData_Concept().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ConceptsExplorationData_Concept().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(ConceptsExplorationData_Concept, a, b);
    }
}
exports.ConceptsExplorationData_Concept = ConceptsExplorationData_Concept;
ConceptsExplorationData_Concept.runtime = protobuf_1.proto3;
ConceptsExplorationData_Concept.typeName = "operand.v1.ConceptsExplorationData.Concept";
ConceptsExplorationData_Concept.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "snippets", kind: "message", T: ConceptsExplorationData_Snippet, repeated: true },
    { no: 2, name: "objects", kind: "map", K: 9 /* ScalarType.STRING */, V: { kind: "message", T: object_pb_js_1.Object$ } },
    { no: 3, name: "summaries", kind: "message", T: ConceptsExplorationData_Summary, repeated: true },
]);
/**
 * @generated from message operand.v1.Exploration
 */
class Exploration extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string public_id = 1;
         */
        this.publicId = "";
        /**
         * @generated from field: operand.v1.ExplorationKind kind = 4;
         */
        this.kind = ExplorationKind.UNSPECIFIED;
        /**
         * @generated from field: operand.v1.ExplorationStatus status = 6;
         */
        this.status = ExplorationStatus.UNSPECIFIED;
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new Exploration().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new Exploration().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new Exploration().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(Exploration, a, b);
    }
}
exports.Exploration = Exploration;
Exploration.runtime = protobuf_1.proto3;
Exploration.typeName = "operand.v1.Exploration";
Exploration.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "public_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "created_at", kind: "message", T: protobuf_1.Timestamp },
    { no: 3, name: "updated_at", kind: "message", T: protobuf_1.Timestamp, opt: true },
    { no: 4, name: "kind", kind: "enum", T: protobuf_1.proto3.getEnumType(ExplorationKind) },
    { no: 5, name: "parameters", kind: "message", T: ExplorationParameters },
    { no: 6, name: "status", kind: "enum", T: protobuf_1.proto3.getEnumType(ExplorationStatus) },
    { no: 7, name: "data", kind: "message", T: ExplorationData, opt: true },
    { no: 8, name: "error_message", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
]);
/**
 * @generated from message operand.v1.ExplorationOptions
 */
class ExplorationOptions extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new ExplorationOptions().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ExplorationOptions().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ExplorationOptions().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(ExplorationOptions, a, b);
    }
}
exports.ExplorationOptions = ExplorationOptions;
ExplorationOptions.runtime = protobuf_1.proto3;
ExplorationOptions.typeName = "operand.v1.ExplorationOptions";
ExplorationOptions.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "include_data", kind: "scalar", T: 8 /* ScalarType.BOOL */, opt: true },
]);
/**
 * @generated from message operand.v1.CreateExplorationRequest
 */
class CreateExplorationRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string index_id = 1;
         */
        this.indexId = "";
        /**
         * @generated from field: operand.v1.ExplorationKind kind = 2;
         */
        this.kind = ExplorationKind.UNSPECIFIED;
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new CreateExplorationRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new CreateExplorationRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new CreateExplorationRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(CreateExplorationRequest, a, b);
    }
}
exports.CreateExplorationRequest = CreateExplorationRequest;
CreateExplorationRequest.runtime = protobuf_1.proto3;
CreateExplorationRequest.typeName = "operand.v1.CreateExplorationRequest";
CreateExplorationRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "index_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "kind", kind: "enum", T: protobuf_1.proto3.getEnumType(ExplorationKind) },
    { no: 3, name: "parameters", kind: "message", T: ExplorationParameters },
    { no: 4, name: "options", kind: "message", T: ExplorationOptions, opt: true },
]);
/**
 * @generated from message operand.v1.CreateExplorationResponse
 */
class CreateExplorationResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new CreateExplorationResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new CreateExplorationResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new CreateExplorationResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(CreateExplorationResponse, a, b);
    }
}
exports.CreateExplorationResponse = CreateExplorationResponse;
CreateExplorationResponse.runtime = protobuf_1.proto3;
CreateExplorationResponse.typeName = "operand.v1.CreateExplorationResponse";
CreateExplorationResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "exploration", kind: "message", T: Exploration },
]);
/**
 * @generated from message operand.v1.DeleteExplorationRequest
 */
class DeleteExplorationRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string public_id = 1;
         */
        this.publicId = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new DeleteExplorationRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new DeleteExplorationRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new DeleteExplorationRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(DeleteExplorationRequest, a, b);
    }
}
exports.DeleteExplorationRequest = DeleteExplorationRequest;
DeleteExplorationRequest.runtime = protobuf_1.proto3;
DeleteExplorationRequest.typeName = "operand.v1.DeleteExplorationRequest";
DeleteExplorationRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "public_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
]);
/**
 * @generated from message operand.v1.DeleteExplorationResponse
 */
class DeleteExplorationResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new DeleteExplorationResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new DeleteExplorationResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new DeleteExplorationResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(DeleteExplorationResponse, a, b);
    }
}
exports.DeleteExplorationResponse = DeleteExplorationResponse;
DeleteExplorationResponse.runtime = protobuf_1.proto3;
DeleteExplorationResponse.typeName = "operand.v1.DeleteExplorationResponse";
DeleteExplorationResponse.fields = protobuf_1.proto3.util.newFieldList(() => []);
/**
 * @generated from message operand.v1.ListExplorationsRequest
 */
class ListExplorationsRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string index_id = 1;
         */
        this.indexId = "";
        /**
         * @generated from field: int32 limit = 2;
         */
        this.limit = 0;
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new ListExplorationsRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ListExplorationsRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ListExplorationsRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(ListExplorationsRequest, a, b);
    }
}
exports.ListExplorationsRequest = ListExplorationsRequest;
ListExplorationsRequest.runtime = protobuf_1.proto3;
ListExplorationsRequest.typeName = "operand.v1.ListExplorationsRequest";
ListExplorationsRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "index_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "limit", kind: "scalar", T: 5 /* ScalarType.INT32 */ },
    { no: 3, name: "offset", kind: "scalar", T: 5 /* ScalarType.INT32 */, opt: true },
    { no: 4, name: "options", kind: "message", T: ExplorationOptions, opt: true },
]);
/**
 * @generated from message operand.v1.ListExplorationsResponse
 */
class ListExplorationsResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: repeated operand.v1.Exploration explorations = 1;
         */
        this.explorations = [];
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new ListExplorationsResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ListExplorationsResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ListExplorationsResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(ListExplorationsResponse, a, b);
    }
}
exports.ListExplorationsResponse = ListExplorationsResponse;
ListExplorationsResponse.runtime = protobuf_1.proto3;
ListExplorationsResponse.typeName = "operand.v1.ListExplorationsResponse";
ListExplorationsResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "explorations", kind: "message", T: Exploration, repeated: true },
]);
/**
 * @generated from message operand.v1.GetExplorationRequest
 */
class GetExplorationRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string public_id = 1;
         */
        this.publicId = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new GetExplorationRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new GetExplorationRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new GetExplorationRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(GetExplorationRequest, a, b);
    }
}
exports.GetExplorationRequest = GetExplorationRequest;
GetExplorationRequest.runtime = protobuf_1.proto3;
GetExplorationRequest.typeName = "operand.v1.GetExplorationRequest";
GetExplorationRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "public_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "options", kind: "message", T: ExplorationOptions, opt: true },
]);
/**
 * @generated from message operand.v1.GetExplorationResponse
 */
class GetExplorationResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new GetExplorationResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new GetExplorationResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new GetExplorationResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(GetExplorationResponse, a, b);
    }
}
exports.GetExplorationResponse = GetExplorationResponse;
GetExplorationResponse.runtime = protobuf_1.proto3;
GetExplorationResponse.typeName = "operand.v1.GetExplorationResponse";
GetExplorationResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "exploration", kind: "message", T: Exploration },
]);
