// @operandinc/sdk/dist/cjs/web/v1/key_pb.js
"use strict";
// @generated by protoc-gen-es v0.1.1 with parameter "target=ts+js"
// @generated from file web/v1/key.proto (package web.v1, syntax proto3)
/* eslint-disable */
/* @ts-nocheck */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteKeyResponse = exports.DeleteKeyRequest = exports.ListKeysResponse = exports.ListKeysRequest = exports.CreateKeyResponse = exports.CreateKeyRequest = exports.APIKey = void 0;
const protobuf_1 = require("@bufbuild/protobuf");
/**
 * @generated from message web.v1.APIKey
 */
class APIKey extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string key = 3;
         */
        this.key = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new APIKey().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new APIKey().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new APIKey().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(APIKey, a, b);
    }
}
exports.APIKey = APIKey;
APIKey.runtime = protobuf_1.proto3;
APIKey.typeName = "web.v1.APIKey";
APIKey.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "created_at", kind: "message", T: protobuf_1.Timestamp },
    { no: 2, name: "last_used", kind: "message", T: protobuf_1.Timestamp, opt: true },
    { no: 3, name: "key", kind: "scalar", T: 9 /* ScalarType.STRING */ },
]);
/**
 * @generated from message web.v1.CreateKeyRequest
 */
class CreateKeyRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new CreateKeyRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new CreateKeyRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new CreateKeyRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(CreateKeyRequest, a, b);
    }
}
exports.CreateKeyRequest = CreateKeyRequest;
CreateKeyRequest.runtime = protobuf_1.proto3;
CreateKeyRequest.typeName = "web.v1.CreateKeyRequest";
CreateKeyRequest.fields = protobuf_1.proto3.util.newFieldList(() => []);
/**
 * @generated from message web.v1.CreateKeyResponse
 */
class CreateKeyResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new CreateKeyResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new CreateKeyResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new CreateKeyResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(CreateKeyResponse, a, b);
    }
}
exports.CreateKeyResponse = CreateKeyResponse;
CreateKeyResponse.runtime = protobuf_1.proto3;
CreateKeyResponse.typeName = "web.v1.CreateKeyResponse";
CreateKeyResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "key", kind: "message", T: APIKey },
]);
/**
 * @generated from message web.v1.ListKeysRequest
 */
class ListKeysRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new ListKeysRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ListKeysRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ListKeysRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(ListKeysRequest, a, b);
    }
}
exports.ListKeysRequest = ListKeysRequest;
ListKeysRequest.runtime = protobuf_1.proto3;
ListKeysRequest.typeName = "web.v1.ListKeysRequest";
ListKeysRequest.fields = protobuf_1.proto3.util.newFieldList(() => []);
/**
 * @generated from message web.v1.ListKeysResponse
 */
class ListKeysResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: repeated web.v1.APIKey keys = 1;
         */
        this.keys = [];
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new ListKeysResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ListKeysResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ListKeysResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(ListKeysResponse, a, b);
    }
}
exports.ListKeysResponse = ListKeysResponse;
ListKeysResponse.runtime = protobuf_1.proto3;
ListKeysResponse.typeName = "web.v1.ListKeysResponse";
ListKeysResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "keys", kind: "message", T: APIKey, repeated: true },
]);
/**
 * @generated from message web.v1.DeleteKeyRequest
 */
class DeleteKeyRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string key = 1;
         */
        this.key = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new DeleteKeyRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new DeleteKeyRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new DeleteKeyRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(DeleteKeyRequest, a, b);
    }
}
exports.DeleteKeyRequest = DeleteKeyRequest;
DeleteKeyRequest.runtime = protobuf_1.proto3;
DeleteKeyRequest.typeName = "web.v1.DeleteKeyRequest";
DeleteKeyRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "key", kind: "scalar", T: 9 /* ScalarType.STRING */ },
]);
/**
 * @generated from message web.v1.DeleteKeyResponse
 */
class DeleteKeyResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new DeleteKeyResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new DeleteKeyResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new DeleteKeyResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(DeleteKeyResponse, a, b);
    }
}
exports.DeleteKeyResponse = DeleteKeyResponse;
DeleteKeyResponse.runtime = protobuf_1.proto3;
DeleteKeyResponse.typeName = "web.v1.DeleteKeyResponse";
DeleteKeyResponse.fields = protobuf_1.proto3.util.newFieldList(() => []);
