// @operandinc/sdk/dist/cjs/web/v1/user_pb.js
"use strict";
// @generated by protoc-gen-es v0.1.1 with parameter "target=ts+js"
// @generated from file web/v1/user.proto (package web.v1, syntax proto3)
/* eslint-disable */
/* @ts-nocheck */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OAuthLinkResponse = exports.OAuthLink = exports.OAuthLinkRequest = exports.ManageBillingResponse = exports.ManageBillingRequest = exports.ConfigureBillingResponse = exports.ConfigureBillingRequest = exports.BillingStatusResponse = exports.BillingStatusRequest = exports.UpdateProfileResponse = exports.UpdateProfileRequest = exports.GetProfileResponse = exports.SocialStats = exports.GetProfileRequest = exports.FinalizeLoginResponse = exports.Token = exports.FinalizeLoginRequest = exports.LoginResponse = exports.LoginRequest = exports.OAuthProvider = exports.BillingPlan = exports.TokenKind = void 0;
const protobuf_1 = require("@bufbuild/protobuf");
const object_pb_js_1 = require("../../operand/v1/object_pb.js");
/**
 * @generated from enum web.v1.TokenKind
 */
var TokenKind;
(function (TokenKind) {
    /**
     * @generated from enum value: TOKEN_KIND_UNSPECIFIED = 0;
     */
    TokenKind[TokenKind["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * @generated from enum value: TOKEN_KIND_LOGIN = 1;
     */
    TokenKind[TokenKind["LOGIN"] = 1] = "LOGIN";
    /**
     * @generated from enum value: TOKEN_KIND_SESSION = 2;
     */
    TokenKind[TokenKind["SESSION"] = 2] = "SESSION";
})(TokenKind = exports.TokenKind || (exports.TokenKind = {}));
// Retrieve enum metadata with: proto3.getEnumType(TokenKind)
protobuf_1.proto3.util.setEnumType(TokenKind, "web.v1.TokenKind", [
    { no: 0, name: "TOKEN_KIND_UNSPECIFIED" },
    { no: 1, name: "TOKEN_KIND_LOGIN" },
    { no: 2, name: "TOKEN_KIND_SESSION" },
]);
/**
 * @generated from enum web.v1.BillingPlan
 */
var BillingPlan;
(function (BillingPlan) {
    /**
     * @generated from enum value: BILLING_PLAN_UNSPECIFIED = 0;
     */
    BillingPlan[BillingPlan["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * @generated from enum value: BILLING_PLAN_FREE = 1;
     */
    BillingPlan[BillingPlan["FREE"] = 1] = "FREE";
    /**
     * @generated from enum value: BILLING_PLAN_PRO = 2;
     */
    BillingPlan[BillingPlan["PRO"] = 2] = "PRO";
    /**
     * @generated from enum value: BILLING_PLAN_TEAMS = 3;
     */
    BillingPlan[BillingPlan["TEAMS"] = 3] = "TEAMS";
    /**
     * @generated from enum value: BILLING_PLAN_ENTERPRISE = 4;
     */
    BillingPlan[BillingPlan["ENTERPRISE"] = 4] = "ENTERPRISE";
})(BillingPlan = exports.BillingPlan || (exports.BillingPlan = {}));
// Retrieve enum metadata with: proto3.getEnumType(BillingPlan)
protobuf_1.proto3.util.setEnumType(BillingPlan, "web.v1.BillingPlan", [
    { no: 0, name: "BILLING_PLAN_UNSPECIFIED" },
    { no: 1, name: "BILLING_PLAN_FREE" },
    { no: 2, name: "BILLING_PLAN_PRO" },
    { no: 3, name: "BILLING_PLAN_TEAMS" },
    { no: 4, name: "BILLING_PLAN_ENTERPRISE" },
]);
/**
 * @generated from enum web.v1.OAuthProvider
 */
var OAuthProvider;
(function (OAuthProvider) {
    /**
     * @generated from enum value: O_AUTH_PROVIDER_UNSPECIFIED = 0;
     */
    OAuthProvider[OAuthProvider["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * @generated from enum value: O_AUTH_PROVIDER_GITHUB = 1;
     */
    OAuthProvider[OAuthProvider["GITHUB"] = 1] = "GITHUB";
})(OAuthProvider = exports.OAuthProvider || (exports.OAuthProvider = {}));
// Retrieve enum metadata with: proto3.getEnumType(OAuthProvider)
protobuf_1.proto3.util.setEnumType(OAuthProvider, "web.v1.OAuthProvider", [
    { no: 0, name: "O_AUTH_PROVIDER_UNSPECIFIED" },
    { no: 1, name: "O_AUTH_PROVIDER_GITHUB" },
]);
/**
 * @generated from message web.v1.LoginRequest
 */
class LoginRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string email = 1;
         */
        this.email = "";
        /**
         * @generated from field: string redirect_url = 2;
         */
        this.redirectUrl = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new LoginRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new LoginRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new LoginRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(LoginRequest, a, b);
    }
}
exports.LoginRequest = LoginRequest;
LoginRequest.runtime = protobuf_1.proto3;
LoginRequest.typeName = "web.v1.LoginRequest";
LoginRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "email", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "redirect_url", kind: "scalar", T: 9 /* ScalarType.STRING */ },
]);
/**
 * @generated from message web.v1.LoginResponse
 */
class LoginResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new LoginResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new LoginResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new LoginResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(LoginResponse, a, b);
    }
}
exports.LoginResponse = LoginResponse;
LoginResponse.runtime = protobuf_1.proto3;
LoginResponse.typeName = "web.v1.LoginResponse";
LoginResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "challenge_sent_at", kind: "message", T: protobuf_1.Timestamp },
    { no: 2, name: "challenge_expires_at", kind: "message", T: protobuf_1.Timestamp },
]);
/**
 * @generated from message web.v1.FinalizeLoginRequest
 */
class FinalizeLoginRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string challenge = 1;
         */
        this.challenge = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new FinalizeLoginRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new FinalizeLoginRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new FinalizeLoginRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(FinalizeLoginRequest, a, b);
    }
}
exports.FinalizeLoginRequest = FinalizeLoginRequest;
FinalizeLoginRequest.runtime = protobuf_1.proto3;
FinalizeLoginRequest.typeName = "web.v1.FinalizeLoginRequest";
FinalizeLoginRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "challenge", kind: "scalar", T: 9 /* ScalarType.STRING */ },
]);
/**
 * @generated from message web.v1.Token
 */
class Token extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string token = 1;
         */
        this.token = "";
        /**
         * @generated from field: web.v1.TokenKind kind = 4;
         */
        this.kind = TokenKind.UNSPECIFIED;
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new Token().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new Token().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new Token().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(Token, a, b);
    }
}
exports.Token = Token;
Token.runtime = protobuf_1.proto3;
Token.typeName = "web.v1.Token";
Token.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "token", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "created_at", kind: "message", T: protobuf_1.Timestamp },
    { no: 3, name: "expires_at", kind: "message", T: protobuf_1.Timestamp },
    { no: 4, name: "kind", kind: "enum", T: protobuf_1.proto3.getEnumType(TokenKind) },
]);
/**
 * @generated from message web.v1.FinalizeLoginResponse
 */
class FinalizeLoginResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new FinalizeLoginResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new FinalizeLoginResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new FinalizeLoginResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(FinalizeLoginResponse, a, b);
    }
}
exports.FinalizeLoginResponse = FinalizeLoginResponse;
FinalizeLoginResponse.runtime = protobuf_1.proto3;
FinalizeLoginResponse.typeName = "web.v1.FinalizeLoginResponse";
FinalizeLoginResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "session", kind: "message", T: Token },
]);
/**
 * @generated from message web.v1.GetProfileRequest
 */
class GetProfileRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new GetProfileRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new GetProfileRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new GetProfileRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(GetProfileRequest, a, b);
    }
}
exports.GetProfileRequest = GetProfileRequest;
GetProfileRequest.runtime = protobuf_1.proto3;
GetProfileRequest.typeName = "web.v1.GetProfileRequest";
GetProfileRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "handle", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 2, name: "include_social_stats", kind: "scalar", T: 8 /* ScalarType.BOOL */, opt: true },
]);
/**
 * @generated from message web.v1.SocialStats
 */
class SocialStats extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: int32 indexes = 1;
         */
        this.indexes = 0;
        /**
         * @generated from field: int32 subscribers = 2;
         */
        this.subscribers = 0;
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new SocialStats().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SocialStats().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SocialStats().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(SocialStats, a, b);
    }
}
exports.SocialStats = SocialStats;
SocialStats.runtime = protobuf_1.proto3;
SocialStats.typeName = "web.v1.SocialStats";
SocialStats.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "indexes", kind: "scalar", T: 5 /* ScalarType.INT32 */ },
    { no: 2, name: "subscribers", kind: "scalar", T: 5 /* ScalarType.INT32 */ },
]);
/**
 * @generated from message web.v1.GetProfileResponse
 */
class GetProfileResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: bool is_current_user = 2;
         */
        this.isCurrentUser = false;
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new GetProfileResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new GetProfileResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new GetProfileResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(GetProfileResponse, a, b);
    }
}
exports.GetProfileResponse = GetProfileResponse;
GetProfileResponse.runtime = protobuf_1.proto3;
GetProfileResponse.typeName = "web.v1.GetProfileResponse";
GetProfileResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "profile", kind: "message", T: object_pb_js_1.UserProfile },
    { no: 2, name: "is_current_user", kind: "scalar", T: 8 /* ScalarType.BOOL */ },
    { no: 3, name: "social_stats", kind: "message", T: SocialStats, opt: true },
]);
/**
 * @generated from message web.v1.UpdateProfileRequest
 */
class UpdateProfileRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new UpdateProfileRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new UpdateProfileRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new UpdateProfileRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(UpdateProfileRequest, a, b);
    }
}
exports.UpdateProfileRequest = UpdateProfileRequest;
UpdateProfileRequest.runtime = protobuf_1.proto3;
UpdateProfileRequest.typeName = "web.v1.UpdateProfileRequest";
UpdateProfileRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "handle", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 2, name: "name", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 3, name: "bio", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 4, name: "avatar_url", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
]);
/**
 * @generated from message web.v1.UpdateProfileResponse
 */
class UpdateProfileResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new UpdateProfileResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new UpdateProfileResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new UpdateProfileResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(UpdateProfileResponse, a, b);
    }
}
exports.UpdateProfileResponse = UpdateProfileResponse;
UpdateProfileResponse.runtime = protobuf_1.proto3;
UpdateProfileResponse.typeName = "web.v1.UpdateProfileResponse";
UpdateProfileResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "profile", kind: "message", T: object_pb_js_1.UserProfile },
]);
/**
 * @generated from message web.v1.BillingStatusRequest
 */
class BillingStatusRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new BillingStatusRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new BillingStatusRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new BillingStatusRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(BillingStatusRequest, a, b);
    }
}
exports.BillingStatusRequest = BillingStatusRequest;
BillingStatusRequest.runtime = protobuf_1.proto3;
BillingStatusRequest.typeName = "web.v1.BillingStatusRequest";
BillingStatusRequest.fields = protobuf_1.proto3.util.newFieldList(() => []);
/**
 * @generated from message web.v1.BillingStatusResponse
 */
class BillingStatusResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string stripe_customer_id = 1;
         */
        this.stripeCustomerId = "";
        /**
         * @generated from field: web.v1.BillingPlan plan = 2;
         */
        this.plan = BillingPlan.UNSPECIFIED;
        /**
         * @generated from field: int32 active_pages = 3;
         */
        this.activePages = 0;
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new BillingStatusResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new BillingStatusResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new BillingStatusResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(BillingStatusResponse, a, b);
    }
}
exports.BillingStatusResponse = BillingStatusResponse;
BillingStatusResponse.runtime = protobuf_1.proto3;
BillingStatusResponse.typeName = "web.v1.BillingStatusResponse";
BillingStatusResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "stripe_customer_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "plan", kind: "enum", T: protobuf_1.proto3.getEnumType(BillingPlan) },
    { no: 3, name: "active_pages", kind: "scalar", T: 5 /* ScalarType.INT32 */ },
    { no: 4, name: "page_limit", kind: "scalar", T: 5 /* ScalarType.INT32 */, opt: true },
]);
/**
 * @generated from message web.v1.ConfigureBillingRequest
 */
class ConfigureBillingRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: web.v1.BillingPlan plan = 1;
         */
        this.plan = BillingPlan.UNSPECIFIED;
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new ConfigureBillingRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ConfigureBillingRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ConfigureBillingRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(ConfigureBillingRequest, a, b);
    }
}
exports.ConfigureBillingRequest = ConfigureBillingRequest;
ConfigureBillingRequest.runtime = protobuf_1.proto3;
ConfigureBillingRequest.typeName = "web.v1.ConfigureBillingRequest";
ConfigureBillingRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "plan", kind: "enum", T: protobuf_1.proto3.getEnumType(BillingPlan) },
    { no: 2, name: "success_url", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 3, name: "cancel_url", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
]);
/**
 * @generated from message web.v1.ConfigureBillingResponse
 */
class ConfigureBillingResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string checkout_link = 1;
         */
        this.checkoutLink = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new ConfigureBillingResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ConfigureBillingResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ConfigureBillingResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(ConfigureBillingResponse, a, b);
    }
}
exports.ConfigureBillingResponse = ConfigureBillingResponse;
ConfigureBillingResponse.runtime = protobuf_1.proto3;
ConfigureBillingResponse.typeName = "web.v1.ConfigureBillingResponse";
ConfigureBillingResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "checkout_link", kind: "scalar", T: 9 /* ScalarType.STRING */ },
]);
/**
 * @generated from message web.v1.ManageBillingRequest
 */
class ManageBillingRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new ManageBillingRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ManageBillingRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ManageBillingRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(ManageBillingRequest, a, b);
    }
}
exports.ManageBillingRequest = ManageBillingRequest;
ManageBillingRequest.runtime = protobuf_1.proto3;
ManageBillingRequest.typeName = "web.v1.ManageBillingRequest";
ManageBillingRequest.fields = protobuf_1.proto3.util.newFieldList(() => []);
/**
 * @generated from message web.v1.ManageBillingResponse
 */
class ManageBillingResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string billing_link = 1;
         */
        this.billingLink = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new ManageBillingResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ManageBillingResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ManageBillingResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(ManageBillingResponse, a, b);
    }
}
exports.ManageBillingResponse = ManageBillingResponse;
ManageBillingResponse.runtime = protobuf_1.proto3;
ManageBillingResponse.typeName = "web.v1.ManageBillingResponse";
ManageBillingResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "billing_link", kind: "scalar", T: 9 /* ScalarType.STRING */ },
]);
/**
 * @generated from message web.v1.OAuthLinkRequest
 */
class OAuthLinkRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: web.v1.OAuthProvider provider = 1;
         */
        this.provider = OAuthProvider.UNSPECIFIED;
        /**
         * @generated from field: string redirect_url = 2;
         */
        this.redirectUrl = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new OAuthLinkRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new OAuthLinkRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new OAuthLinkRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(OAuthLinkRequest, a, b);
    }
}
exports.OAuthLinkRequest = OAuthLinkRequest;
OAuthLinkRequest.runtime = protobuf_1.proto3;
OAuthLinkRequest.typeName = "web.v1.OAuthLinkRequest";
OAuthLinkRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "provider", kind: "enum", T: protobuf_1.proto3.getEnumType(OAuthProvider) },
    { no: 2, name: "redirect_url", kind: "scalar", T: 9 /* ScalarType.STRING */ },
]);
/**
 * @generated from message web.v1.OAuthLink
 */
class OAuthLink extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string public_id = 1;
         */
        this.publicId = "";
        /**
         * @generated from field: web.v1.OAuthProvider provider = 2;
         */
        this.provider = OAuthProvider.UNSPECIFIED;
        /**
         * Note: We don't return refresh_token or expires_at here, since it's purely an internal thing.
         *
         * @generated from field: string access_token = 4;
         */
        this.accessToken = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new OAuthLink().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new OAuthLink().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new OAuthLink().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(OAuthLink, a, b);
    }
}
exports.OAuthLink = OAuthLink;
OAuthLink.runtime = protobuf_1.proto3;
OAuthLink.typeName = "web.v1.OAuthLink";
OAuthLink.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "public_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "provider", kind: "enum", T: protobuf_1.proto3.getEnumType(OAuthProvider) },
    { no: 3, name: "created_at", kind: "message", T: protobuf_1.Timestamp },
    { no: 4, name: "access_token", kind: "scalar", T: 9 /* ScalarType.STRING */ },
]);
/**
 * @generated from message web.v1.OAuthLinkResponse
 */
class OAuthLinkResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from oneof web.v1.OAuthLinkResponse.resp
         */
        this.resp = { case: undefined };
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new OAuthLinkResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new OAuthLinkResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new OAuthLinkResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(OAuthLinkResponse, a, b);
    }
}
exports.OAuthLinkResponse = OAuthLinkResponse;
OAuthLinkResponse.runtime = protobuf_1.proto3;
OAuthLinkResponse.typeName = "web.v1.OAuthLinkResponse";
OAuthLinkResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "setup_url", kind: "scalar", T: 9 /* ScalarType.STRING */, oneof: "resp" },
    { no: 2, name: "existing", kind: "message", T: OAuthLink, oneof: "resp" },
]);
