// @operandinc/sdk/dist/cjs/operand/v1/brain_pb.js
"use strict";
// @generated by protoc-gen-es v0.1.1 with parameter "target=ts+js"
// @generated from file operand/v1/brain.proto (package operand.v1, syntax proto3)
/* eslint-disable */
/* @ts-nocheck */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MigrateResponse = exports.MigrateRequest = exports.RequiresMigrationResponse = exports.RequiresMigrationRequest = void 0;
const protobuf_1 = require("@bufbuild/protobuf");
/**
 * RequiresMigrationRequest is the request for the RequiresMigration RPC.
 *
 * @generated from message operand.v1.RequiresMigrationRequest
 */
class RequiresMigrationRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new RequiresMigrationRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new RequiresMigrationRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new RequiresMigrationRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(RequiresMigrationRequest, a, b);
    }
}
exports.RequiresMigrationRequest = RequiresMigrationRequest;
RequiresMigrationRequest.runtime = protobuf_1.proto3;
RequiresMigrationRequest.typeName = "operand.v1.RequiresMigrationRequest";
RequiresMigrationRequest.fields = protobuf_1.proto3.util.newFieldList(() => []);
/**
 * RequiresMigrationResponse is the response for the RequiresMigration RPC.
 *
 * @generated from message operand.v1.RequiresMigrationResponse
 */
class RequiresMigrationResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: bool requires_migration = 1;
         */
        this.requiresMigration = false;
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new RequiresMigrationResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new RequiresMigrationResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new RequiresMigrationResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(RequiresMigrationResponse, a, b);
    }
}
exports.RequiresMigrationResponse = RequiresMigrationResponse;
RequiresMigrationResponse.runtime = protobuf_1.proto3;
RequiresMigrationResponse.typeName = "operand.v1.RequiresMigrationResponse";
RequiresMigrationResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "requires_migration", kind: "scalar", T: 8 /* ScalarType.BOOL */ },
]);
/**
 * MigrateRequest is the request for the Migrate RPC.
 *
 * @generated from message operand.v1.MigrateRequest
 */
class MigrateRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string destination_index_id = 1;
         */
        this.destinationIndexId = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new MigrateRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new MigrateRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new MigrateRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(MigrateRequest, a, b);
    }
}
exports.MigrateRequest = MigrateRequest;
MigrateRequest.runtime = protobuf_1.proto3;
MigrateRequest.typeName = "operand.v1.MigrateRequest";
MigrateRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "destination_index_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
]);
/**
 * MigrateResponse is the response for the Migrate RPC.
 *
 * @generated from message operand.v1.MigrateResponse
 */
class MigrateResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new MigrateResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new MigrateResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new MigrateResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(MigrateResponse, a, b);
    }
}
exports.MigrateResponse = MigrateResponse;
MigrateResponse.runtime = protobuf_1.proto3;
MigrateResponse.typeName = "operand.v1.MigrateResponse";
MigrateResponse.fields = protobuf_1.proto3.util.newFieldList(() => []);
