// @operandinc/sdk/dist/cjs/operand/v1/notification_pb.js
"use strict";
// @generated by protoc-gen-es v0.1.1 with parameter "target=ts+js"
// @generated from file operand/v1/notification.proto (package operand.v1, syntax proto3)
/* eslint-disable */
/* @ts-nocheck */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateSettingsResponse = exports.UpdateSettingsRequest = exports.SettingsResponse = exports.SettingsRequest = exports.NotificationSettings = exports.AcknowledgeResponse = exports.AcknowledgeRequest = exports.PendingResponse = exports.Notification = exports.NotificationAction_RedirectTo = exports.NotificationAction_None = exports.NotificationAction = exports.PendingRequest = void 0;
const protobuf_1 = require("@bufbuild/protobuf");
/**
 * PendingRequest is the request for the Pending method.
 *
 * @generated from message operand.v1.PendingRequest
 */
class PendingRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: int32 limit = 1;
         */
        this.limit = 0;
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new PendingRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new PendingRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new PendingRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(PendingRequest, a, b);
    }
}
exports.PendingRequest = PendingRequest;
PendingRequest.runtime = protobuf_1.proto3;
PendingRequest.typeName = "operand.v1.PendingRequest";
PendingRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "limit", kind: "scalar", T: 5 /* ScalarType.INT32 */ },
    { no: 2, name: "cursor", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
]);
/**
 * NotificationAction contains an action that can be taken on a notification.
 *
 * @generated from message operand.v1.NotificationAction
 */
class NotificationAction extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from oneof operand.v1.NotificationAction.action
         */
        this.action = { case: undefined };
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new NotificationAction().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new NotificationAction().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new NotificationAction().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(NotificationAction, a, b);
    }
}
exports.NotificationAction = NotificationAction;
NotificationAction.runtime = protobuf_1.proto3;
NotificationAction.typeName = "operand.v1.NotificationAction";
NotificationAction.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "none", kind: "message", T: NotificationAction_None, oneof: "action" },
    { no: 2, name: "redirect_to", kind: "message", T: NotificationAction_RedirectTo, oneof: "action" },
]);
/**
 * @generated from message operand.v1.NotificationAction.None
 */
class NotificationAction_None extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new NotificationAction_None().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new NotificationAction_None().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new NotificationAction_None().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(NotificationAction_None, a, b);
    }
}
exports.NotificationAction_None = NotificationAction_None;
NotificationAction_None.runtime = protobuf_1.proto3;
NotificationAction_None.typeName = "operand.v1.NotificationAction.None";
NotificationAction_None.fields = protobuf_1.proto3.util.newFieldList(() => []);
/**
 * @generated from message operand.v1.NotificationAction.RedirectTo
 */
class NotificationAction_RedirectTo extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string url = 1;
         */
        this.url = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new NotificationAction_RedirectTo().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new NotificationAction_RedirectTo().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new NotificationAction_RedirectTo().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(NotificationAction_RedirectTo, a, b);
    }
}
exports.NotificationAction_RedirectTo = NotificationAction_RedirectTo;
NotificationAction_RedirectTo.runtime = protobuf_1.proto3;
NotificationAction_RedirectTo.typeName = "operand.v1.NotificationAction.RedirectTo";
NotificationAction_RedirectTo.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "url", kind: "scalar", T: 9 /* ScalarType.STRING */ },
]);
/**
 * Notification is a notification sent to a user.
 *
 * @generated from message operand.v1.Notification
 */
class Notification extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string id = 1;
         */
        this.id = "";
        /**
         * @generated from field: string title = 3;
         */
        this.title = "";
        /**
         * @generated from field: string body = 4;
         */
        this.body = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new Notification().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new Notification().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new Notification().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(Notification, a, b);
    }
}
exports.Notification = Notification;
Notification.runtime = protobuf_1.proto3;
Notification.typeName = "operand.v1.Notification";
Notification.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "created_at", kind: "message", T: protobuf_1.Timestamp },
    { no: 3, name: "title", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 4, name: "body", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 5, name: "action", kind: "message", T: NotificationAction },
]);
/**
 * PendingResponse is the response for the Pending method.
 *
 * @generated from message operand.v1.PendingResponse
 */
class PendingResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: repeated operand.v1.Notification notifications = 1;
         */
        this.notifications = [];
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new PendingResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new PendingResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new PendingResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(PendingResponse, a, b);
    }
}
exports.PendingResponse = PendingResponse;
PendingResponse.runtime = protobuf_1.proto3;
PendingResponse.typeName = "operand.v1.PendingResponse";
PendingResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "notifications", kind: "message", T: Notification, repeated: true },
    { no: 2, name: "next_cursor", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
]);
/**
 * AcknowledgeRequest is the request for the Acknowledge method.
 *
 * @generated from message operand.v1.AcknowledgeRequest
 */
class AcknowledgeRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: repeated string ids = 1;
         */
        this.ids = [];
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new AcknowledgeRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new AcknowledgeRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new AcknowledgeRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(AcknowledgeRequest, a, b);
    }
}
exports.AcknowledgeRequest = AcknowledgeRequest;
AcknowledgeRequest.runtime = protobuf_1.proto3;
AcknowledgeRequest.typeName = "operand.v1.AcknowledgeRequest";
AcknowledgeRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "ids", kind: "scalar", T: 9 /* ScalarType.STRING */, repeated: true },
]);
/**
 * AcknowledgeResponse is the response for the Acknowledge method.
 *
 * @generated from message operand.v1.AcknowledgeResponse
 */
class AcknowledgeResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new AcknowledgeResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new AcknowledgeResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new AcknowledgeResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(AcknowledgeResponse, a, b);
    }
}
exports.AcknowledgeResponse = AcknowledgeResponse;
AcknowledgeResponse.runtime = protobuf_1.proto3;
AcknowledgeResponse.typeName = "operand.v1.AcknowledgeResponse";
AcknowledgeResponse.fields = protobuf_1.proto3.util.newFieldList(() => []);
/**
 * NotificationSettings contains the notification settings for a user.
 *
 * @generated from message operand.v1.NotificationSettings
 */
class NotificationSettings extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: bool disable_emails = 1;
         */
        this.disableEmails = false;
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new NotificationSettings().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new NotificationSettings().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new NotificationSettings().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(NotificationSettings, a, b);
    }
}
exports.NotificationSettings = NotificationSettings;
NotificationSettings.runtime = protobuf_1.proto3;
NotificationSettings.typeName = "operand.v1.NotificationSettings";
NotificationSettings.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "disable_emails", kind: "scalar", T: 8 /* ScalarType.BOOL */ },
]);
/**
 * SettingsRequest is the request for the Settings method.
 *
 * @generated from message operand.v1.SettingsRequest
 */
class SettingsRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new SettingsRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SettingsRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SettingsRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(SettingsRequest, a, b);
    }
}
exports.SettingsRequest = SettingsRequest;
SettingsRequest.runtime = protobuf_1.proto3;
SettingsRequest.typeName = "operand.v1.SettingsRequest";
SettingsRequest.fields = protobuf_1.proto3.util.newFieldList(() => []);
/**
 * SettingsResponse is the response for the Settings method.
 *
 * @generated from message operand.v1.SettingsResponse
 */
class SettingsResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new SettingsResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SettingsResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SettingsResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(SettingsResponse, a, b);
    }
}
exports.SettingsResponse = SettingsResponse;
SettingsResponse.runtime = protobuf_1.proto3;
SettingsResponse.typeName = "operand.v1.SettingsResponse";
SettingsResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "settings", kind: "message", T: NotificationSettings },
]);
/**
 * UpdateSettingsRequest is the request for the UpdateSettings method.
 *
 * @generated from message operand.v1.UpdateSettingsRequest
 */
class UpdateSettingsRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new UpdateSettingsRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new UpdateSettingsRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new UpdateSettingsRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(UpdateSettingsRequest, a, b);
    }
}
exports.UpdateSettingsRequest = UpdateSettingsRequest;
UpdateSettingsRequest.runtime = protobuf_1.proto3;
UpdateSettingsRequest.typeName = "operand.v1.UpdateSettingsRequest";
UpdateSettingsRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "settings", kind: "message", T: NotificationSettings },
]);
/**
 * UpdateSettingsResponse is the response for the UpdateSettings method.
 *
 * @generated from message operand.v1.UpdateSettingsResponse
 */
class UpdateSettingsResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new UpdateSettingsResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new UpdateSettingsResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new UpdateSettingsResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(UpdateSettingsResponse, a, b);
    }
}
exports.UpdateSettingsResponse = UpdateSettingsResponse;
UpdateSettingsResponse.runtime = protobuf_1.proto3;
UpdateSettingsResponse.typeName = "operand.v1.UpdateSettingsResponse";
UpdateSettingsResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "settings", kind: "message", T: NotificationSettings },
]);
