// @operandinc/sdk/dist/cjs/operand/v1/notification_connectweb.js
"use strict";
// @generated by protoc-gen-connect-web v0.2.1 with parameter "target=ts"
// @generated from file operand/v1/notification.proto (package operand.v1, syntax proto3)
/* eslint-disable */
/* @ts-nocheck */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotificationService = void 0;
const notification_pb_js_1 = require("./notification_pb.js");
const protobuf_1 = require("@bufbuild/protobuf");
/**
 * NotificationService manages notifications sent to users. Users are notified
 * when things happen that are relevant to them, and they can configure notification
 * settings (i.e. emails, push notifications, etc.) to control how they are notified.
 * This service authentication via API key, set in the `Authorization` header.
 *
 * @generated from service operand.v1.NotificationService
 */
exports.NotificationService = {
    typeName: "operand.v1.NotificationService",
    methods: {
        /**
         * Pending returns a list of pending notifications for the user.
         *
         * @generated from rpc operand.v1.NotificationService.Pending
         */
        pending: {
            name: "Pending",
            I: notification_pb_js_1.PendingRequest,
            O: notification_pb_js_1.PendingResponse,
            kind: protobuf_1.MethodKind.Unary,
        },
        /**
         * Acknowledge acknowledges one or more notifications.
         *
         * @generated from rpc operand.v1.NotificationService.Acknowledge
         */
        acknowledge: {
            name: "Acknowledge",
            I: notification_pb_js_1.AcknowledgeRequest,
            O: notification_pb_js_1.AcknowledgeResponse,
            kind: protobuf_1.MethodKind.Unary,
        },
        /**
         * Settings returns the notification settings for the user.
         *
         * @generated from rpc operand.v1.NotificationService.Settings
         */
        settings: {
            name: "Settings",
            I: notification_pb_js_1.SettingsRequest,
            O: notification_pb_js_1.SettingsResponse,
            kind: protobuf_1.MethodKind.Unary,
        },
        /**
         * UpdateSettings updates the notification settings for the user.
         *
         * @generated from rpc operand.v1.NotificationService.UpdateSettings
         */
        updateSettings: {
            name: "UpdateSettings",
            I: notification_pb_js_1.UpdateSettingsRequest,
            O: notification_pb_js_1.UpdateSettingsResponse,
            kind: protobuf_1.MethodKind.Unary,
        },
    }
};
