// @operandinc/sdk/dist/cjs/operand/v1/object_pb.js
"use strict";
// @generated by protoc-gen-es v0.1.1 with parameter "target=ts+js"
// @generated from file operand/v1/object.proto (package operand.v1, syntax proto3)
/* eslint-disable */
/* @ts-nocheck */
Object.defineProperty(exports, "__esModule", { value: true });
exports.KeyedProperty = exports.Condition = exports.NotCondition = exports.Filter = exports.CountResponse = exports.CountRequest = exports.DeleteResponse = exports.DeleteRequest = exports.GetResponse = exports.GetRequest = exports.ListResponse = exports.ListRequest = exports.SuggestionsResponse_AnnotatedUpsert = exports.SuggestionsResponse = exports.SuggestionsRequest = exports.UpsertResponse = exports.UpsertRequest = exports.ObjectOptions = exports.Object$ = exports.IndexingProgress = exports.ObjectPreview = exports.Author = exports.UserProfile = exports.NumberArray = exports.TextArray = exports.Property = exports.Properties = exports.MeetingBotMetadata = exports.LinearMetadata = exports.DiscordMetadata = exports.NotionMetadata = exports.SourceCodeMetadata = exports.GithubRepositoryMetadata = exports.SlackMetadata = exports.YouTubeMetadata = exports.EPUBMetadata = exports.PDFMetadata = exports.SitemapMetadata = exports.PodcastMetadata = exports.AudioMetadata = exports.RSSMetadata = exports.MarkdownMetadata = exports.HTMLMetadata = exports.TextMetadata = exports.CollectionMetadata = exports.ObjectMetadata = exports.ObjectStatus = exports.ProgrammingLanguage = exports.AudioFileExt = exports.ObjectType = void 0;
exports.AnswerResponse = exports.AnswerRequest = exports.Answer = exports.SearchWithinResponse_Match = exports.SearchWithinResponse = exports.SearchWithinRequest = exports.Range = void 0;
const protobuf_1 = require("@bufbuild/protobuf");
/**
 * @generated from enum operand.v1.ObjectType
 */
var ObjectType;
(function (ObjectType) {
    /**
     * @generated from enum value: OBJECT_TYPE_UNSPECIFIED = 0;
     */
    ObjectType[ObjectType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * @generated from enum value: OBJECT_TYPE_COLLECTION = 1;
     */
    ObjectType[ObjectType["COLLECTION"] = 1] = "COLLECTION";
    /**
     * @generated from enum value: OBJECT_TYPE_TEXT = 2;
     */
    ObjectType[ObjectType["TEXT"] = 2] = "TEXT";
    /**
     * @generated from enum value: OBJECT_TYPE_HTML = 3;
     */
    ObjectType[ObjectType["HTML"] = 3] = "HTML";
    /**
     * @generated from enum value: OBJECT_TYPE_RSS = 4;
     */
    ObjectType[ObjectType["RSS"] = 4] = "RSS";
    /**
     * @generated from enum value: OBJECT_TYPE_AUDIO = 5;
     */
    ObjectType[ObjectType["AUDIO"] = 5] = "AUDIO";
    /**
     * @generated from enum value: OBJECT_TYPE_PODCAST = 6;
     */
    ObjectType[ObjectType["PODCAST"] = 6] = "PODCAST";
    /**
     * @generated from enum value: OBJECT_TYPE_SITEMAP = 7;
     */
    ObjectType[ObjectType["SITEMAP"] = 7] = "SITEMAP";
    /**
     * @generated from enum value: OBJECT_TYPE_PDF = 8;
     */
    ObjectType[ObjectType["PDF"] = 8] = "PDF";
    /**
     * @generated from enum value: OBJECT_TYPE_EPUB = 9;
     */
    ObjectType[ObjectType["EPUB"] = 9] = "EPUB";
    /**
     * @generated from enum value: OBJECT_TYPE_YOUTUBE = 10;
     */
    ObjectType[ObjectType["YOUTUBE"] = 10] = "YOUTUBE";
    /**
     * @generated from enum value: OBJECT_TYPE_MARKDOWN = 11;
     */
    ObjectType[ObjectType["MARKDOWN"] = 11] = "MARKDOWN";
    /**
     * @generated from enum value: OBJECT_TYPE_SLACK = 12;
     */
    ObjectType[ObjectType["SLACK"] = 12] = "SLACK";
    /**
     * @generated from enum value: OBJECT_TYPE_GITHUB_REPOSITORY = 13;
     */
    ObjectType[ObjectType["GITHUB_REPOSITORY"] = 13] = "GITHUB_REPOSITORY";
    /**
     * @generated from enum value: OBJECT_TYPE_SOURCE_CODE = 14;
     */
    ObjectType[ObjectType["SOURCE_CODE"] = 14] = "SOURCE_CODE";
    /**
     * @generated from enum value: OBJECT_TYPE_NOTION = 15;
     */
    ObjectType[ObjectType["NOTION"] = 15] = "NOTION";
    /**
     * @generated from enum value: OBJECT_TYPE_DISCORD = 16;
     */
    ObjectType[ObjectType["DISCORD"] = 16] = "DISCORD";
    /**
     * @generated from enum value: OBJECT_TYPE_LINEAR = 17;
     */
    ObjectType[ObjectType["LINEAR"] = 17] = "LINEAR";
    /**
     * @generated from enum value: OBJECT_TYPE_MEETING_BOT = 18;
     */
    ObjectType[ObjectType["MEETING_BOT"] = 18] = "MEETING_BOT";
})(ObjectType = exports.ObjectType || (exports.ObjectType = {}));
// Retrieve enum metadata with: proto3.getEnumType(ObjectType)
protobuf_1.proto3.util.setEnumType(ObjectType, "operand.v1.ObjectType", [
    { no: 0, name: "OBJECT_TYPE_UNSPECIFIED" },
    { no: 1, name: "OBJECT_TYPE_COLLECTION" },
    { no: 2, name: "OBJECT_TYPE_TEXT" },
    { no: 3, name: "OBJECT_TYPE_HTML" },
    { no: 4, name: "OBJECT_TYPE_RSS" },
    { no: 5, name: "OBJECT_TYPE_AUDIO" },
    { no: 6, name: "OBJECT_TYPE_PODCAST" },
    { no: 7, name: "OBJECT_TYPE_SITEMAP" },
    { no: 8, name: "OBJECT_TYPE_PDF" },
    { no: 9, name: "OBJECT_TYPE_EPUB" },
    { no: 10, name: "OBJECT_TYPE_YOUTUBE" },
    { no: 11, name: "OBJECT_TYPE_MARKDOWN" },
    { no: 12, name: "OBJECT_TYPE_SLACK" },
    { no: 13, name: "OBJECT_TYPE_GITHUB_REPOSITORY" },
    { no: 14, name: "OBJECT_TYPE_SOURCE_CODE" },
    { no: 15, name: "OBJECT_TYPE_NOTION" },
    { no: 16, name: "OBJECT_TYPE_DISCORD" },
    { no: 17, name: "OBJECT_TYPE_LINEAR" },
    { no: 18, name: "OBJECT_TYPE_MEETING_BOT" },
]);
/**
 * @generated from enum operand.v1.AudioFileExt
 */
var AudioFileExt;
(function (AudioFileExt) {
    /**
     * @generated from enum value: AUDIO_FILE_EXT_UNSPECIFIED = 0;
     */
    AudioFileExt[AudioFileExt["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * @generated from enum value: AUDIO_FILE_EXT_MP3 = 1;
     */
    AudioFileExt[AudioFileExt["MP3"] = 1] = "MP3";
    /**
     * @generated from enum value: AUDIO_FILE_EXT_OGG = 2;
     */
    AudioFileExt[AudioFileExt["OGG"] = 2] = "OGG";
    /**
     * @generated from enum value: AUDIO_FILE_EXT_FLAC = 3;
     */
    AudioFileExt[AudioFileExt["FLAC"] = 3] = "FLAC";
    /**
     * @generated from enum value: AUDIO_FILE_EXT_WAV = 4;
     */
    AudioFileExt[AudioFileExt["WAV"] = 4] = "WAV";
    /**
     * @generated from enum value: AUDIO_FILE_EXT_MP4 = 5;
     */
    AudioFileExt[AudioFileExt["MP4"] = 5] = "MP4";
})(AudioFileExt = exports.AudioFileExt || (exports.AudioFileExt = {}));
// Retrieve enum metadata with: proto3.getEnumType(AudioFileExt)
protobuf_1.proto3.util.setEnumType(AudioFileExt, "operand.v1.AudioFileExt", [
    { no: 0, name: "AUDIO_FILE_EXT_UNSPECIFIED" },
    { no: 1, name: "AUDIO_FILE_EXT_MP3" },
    { no: 2, name: "AUDIO_FILE_EXT_OGG" },
    { no: 3, name: "AUDIO_FILE_EXT_FLAC" },
    { no: 4, name: "AUDIO_FILE_EXT_WAV" },
    { no: 5, name: "AUDIO_FILE_EXT_MP4" },
]);
/**
 * @generated from enum operand.v1.ProgrammingLanguage
 */
var ProgrammingLanguage;
(function (ProgrammingLanguage) {
    /**
     * @generated from enum value: PROGRAMMING_LANGUAGE_UNSPECIFIED = 0;
     */
    ProgrammingLanguage[ProgrammingLanguage["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * @generated from enum value: PROGRAMMING_LANGUAGE_GO = 1;
     */
    ProgrammingLanguage[ProgrammingLanguage["GO"] = 1] = "GO";
    /**
     * @generated from enum value: PROGRAMMING_LANGUAGE_PYTHON = 2;
     */
    ProgrammingLanguage[ProgrammingLanguage["PYTHON"] = 2] = "PYTHON";
})(ProgrammingLanguage = exports.ProgrammingLanguage || (exports.ProgrammingLanguage = {}));
// Retrieve enum metadata with: proto3.getEnumType(ProgrammingLanguage)
protobuf_1.proto3.util.setEnumType(ProgrammingLanguage, "operand.v1.ProgrammingLanguage", [
    { no: 0, name: "PROGRAMMING_LANGUAGE_UNSPECIFIED" },
    { no: 1, name: "PROGRAMMING_LANGUAGE_GO" },
    { no: 2, name: "PROGRAMMING_LANGUAGE_PYTHON" },
]);
/**
 * @generated from enum operand.v1.ObjectStatus
 */
var ObjectStatus;
(function (ObjectStatus) {
    /**
     * @generated from enum value: OBJECT_STATUS_UNSPECIFIED = 0;
     */
    ObjectStatus[ObjectStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * @generated from enum value: OBJECT_STATUS_QUEUED = 1;
     */
    ObjectStatus[ObjectStatus["QUEUED"] = 1] = "QUEUED";
    /**
     * @generated from enum value: OBJECT_STATUS_INDEXING = 2;
     */
    ObjectStatus[ObjectStatus["INDEXING"] = 2] = "INDEXING";
    /**
     * @generated from enum value: OBJECT_STATUS_READY = 3;
     */
    ObjectStatus[ObjectStatus["READY"] = 3] = "READY";
    /**
     * @generated from enum value: OBJECT_STATUS_ERROR = 4;
     */
    ObjectStatus[ObjectStatus["ERROR"] = 4] = "ERROR";
})(ObjectStatus = exports.ObjectStatus || (exports.ObjectStatus = {}));
// Retrieve enum metadata with: proto3.getEnumType(ObjectStatus)
protobuf_1.proto3.util.setEnumType(ObjectStatus, "operand.v1.ObjectStatus", [
    { no: 0, name: "OBJECT_STATUS_UNSPECIFIED" },
    { no: 1, name: "OBJECT_STATUS_QUEUED" },
    { no: 2, name: "OBJECT_STATUS_INDEXING" },
    { no: 3, name: "OBJECT_STATUS_READY" },
    { no: 4, name: "OBJECT_STATUS_ERROR" },
]);
/**
 * @generated from message operand.v1.ObjectMetadata
 */
class ObjectMetadata extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from oneof operand.v1.ObjectMetadata.value
         */
        this.value = { case: undefined };
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new ObjectMetadata().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ObjectMetadata().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ObjectMetadata().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(ObjectMetadata, a, b);
    }
}
exports.ObjectMetadata = ObjectMetadata;
ObjectMetadata.runtime = protobuf_1.proto3;
ObjectMetadata.typeName = "operand.v1.ObjectMetadata";
ObjectMetadata.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "collection", kind: "message", T: CollectionMetadata, oneof: "value" },
    { no: 2, name: "text", kind: "message", T: TextMetadata, oneof: "value" },
    { no: 3, name: "html", kind: "message", T: HTMLMetadata, oneof: "value" },
    { no: 4, name: "rss", kind: "message", T: RSSMetadata, oneof: "value" },
    { no: 5, name: "audio", kind: "message", T: AudioMetadata, oneof: "value" },
    { no: 6, name: "podcast", kind: "message", T: PodcastMetadata, oneof: "value" },
    { no: 7, name: "sitemap", kind: "message", T: SitemapMetadata, oneof: "value" },
    { no: 8, name: "pdf", kind: "message", T: PDFMetadata, oneof: "value" },
    { no: 9, name: "epub", kind: "message", T: EPUBMetadata, oneof: "value" },
    { no: 10, name: "youtube", kind: "message", T: YouTubeMetadata, oneof: "value" },
    { no: 11, name: "markdown", kind: "message", T: MarkdownMetadata, oneof: "value" },
    { no: 12, name: "slack", kind: "message", T: SlackMetadata, oneof: "value" },
    { no: 13, name: "github", kind: "message", T: GithubRepositoryMetadata, oneof: "value" },
    { no: 14, name: "source_code", kind: "message", T: SourceCodeMetadata, oneof: "value" },
    { no: 15, name: "notion", kind: "message", T: NotionMetadata, oneof: "value" },
    { no: 16, name: "discord", kind: "message", T: DiscordMetadata, oneof: "value" },
    { no: 17, name: "linear", kind: "message", T: LinearMetadata, oneof: "value" },
    { no: 18, name: "meeting_bot", kind: "message", T: MeetingBotMetadata, oneof: "value" },
]);
/**
 * @generated from message operand.v1.CollectionMetadata
 */
class CollectionMetadata extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new CollectionMetadata().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new CollectionMetadata().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new CollectionMetadata().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(CollectionMetadata, a, b);
    }
}
exports.CollectionMetadata = CollectionMetadata;
CollectionMetadata.runtime = protobuf_1.proto3;
CollectionMetadata.typeName = "operand.v1.CollectionMetadata";
CollectionMetadata.fields = protobuf_1.proto3.util.newFieldList(() => []);
/**
 * @generated from message operand.v1.TextMetadata
 */
class TextMetadata extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string text = 1;
         */
        this.text = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new TextMetadata().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new TextMetadata().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new TextMetadata().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(TextMetadata, a, b);
    }
}
exports.TextMetadata = TextMetadata;
TextMetadata.runtime = protobuf_1.proto3;
TextMetadata.typeName = "operand.v1.TextMetadata";
TextMetadata.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "text", kind: "scalar", T: 9 /* ScalarType.STRING */ },
]);
/**
 * @generated from message operand.v1.HTMLMetadata
 */
class HTMLMetadata extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new HTMLMetadata().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new HTMLMetadata().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new HTMLMetadata().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(HTMLMetadata, a, b);
    }
}
exports.HTMLMetadata = HTMLMetadata;
HTMLMetadata.runtime = protobuf_1.proto3;
HTMLMetadata.typeName = "operand.v1.HTMLMetadata";
HTMLMetadata.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "html", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
]);
/**
 * @generated from message operand.v1.MarkdownMetadata
 */
class MarkdownMetadata extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new MarkdownMetadata().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new MarkdownMetadata().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new MarkdownMetadata().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(MarkdownMetadata, a, b);
    }
}
exports.MarkdownMetadata = MarkdownMetadata;
MarkdownMetadata.runtime = protobuf_1.proto3;
MarkdownMetadata.typeName = "operand.v1.MarkdownMetadata";
MarkdownMetadata.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "markdown", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
]);
/**
 * The URL is stored in properties, not here.
 *
 * @generated from message operand.v1.RSSMetadata
 */
class RSSMetadata extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new RSSMetadata().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new RSSMetadata().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new RSSMetadata().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(RSSMetadata, a, b);
    }
}
exports.RSSMetadata = RSSMetadata;
RSSMetadata.runtime = protobuf_1.proto3;
RSSMetadata.typeName = "operand.v1.RSSMetadata";
RSSMetadata.fields = protobuf_1.proto3.util.newFieldList(() => []);
/**
 * If the _url property is set, this metadata is auto-populated.
 *
 * @generated from message operand.v1.AudioMetadata
 */
class AudioMetadata extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new AudioMetadata().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new AudioMetadata().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new AudioMetadata().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(AudioMetadata, a, b);
    }
}
exports.AudioMetadata = AudioMetadata;
AudioMetadata.runtime = protobuf_1.proto3;
AudioMetadata.typeName = "operand.v1.AudioMetadata";
AudioMetadata.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "file_ext", kind: "enum", T: protobuf_1.proto3.getEnumType(AudioFileExt), opt: true },
    { no: 2, name: "audio", kind: "scalar", T: 12 /* ScalarType.BYTES */, opt: true },
]);
/**
 * @generated from message operand.v1.PodcastMetadata
 */
class PodcastMetadata extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string listennotes_id = 1;
         */
        this.listennotesId = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new PodcastMetadata().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new PodcastMetadata().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new PodcastMetadata().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(PodcastMetadata, a, b);
    }
}
exports.PodcastMetadata = PodcastMetadata;
PodcastMetadata.runtime = protobuf_1.proto3;
PodcastMetadata.typeName = "operand.v1.PodcastMetadata";
PodcastMetadata.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "listennotes_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
]);
/**
 * @generated from message operand.v1.SitemapMetadata
 */
class SitemapMetadata extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new SitemapMetadata().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SitemapMetadata().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SitemapMetadata().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(SitemapMetadata, a, b);
    }
}
exports.SitemapMetadata = SitemapMetadata;
SitemapMetadata.runtime = protobuf_1.proto3;
SitemapMetadata.typeName = "operand.v1.SitemapMetadata";
SitemapMetadata.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "url_regex", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
]);
/**
 * @generated from message operand.v1.PDFMetadata
 */
class PDFMetadata extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new PDFMetadata().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new PDFMetadata().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new PDFMetadata().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(PDFMetadata, a, b);
    }
}
exports.PDFMetadata = PDFMetadata;
PDFMetadata.runtime = protobuf_1.proto3;
PDFMetadata.typeName = "operand.v1.PDFMetadata";
PDFMetadata.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "pdf", kind: "scalar", T: 12 /* ScalarType.BYTES */, opt: true },
]);
/**
 * @generated from message operand.v1.EPUBMetadata
 */
class EPUBMetadata extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new EPUBMetadata().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new EPUBMetadata().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new EPUBMetadata().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(EPUBMetadata, a, b);
    }
}
exports.EPUBMetadata = EPUBMetadata;
EPUBMetadata.runtime = protobuf_1.proto3;
EPUBMetadata.typeName = "operand.v1.EPUBMetadata";
EPUBMetadata.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "epub", kind: "scalar", T: 12 /* ScalarType.BYTES */, opt: true },
]);
/**
 * The URL is stored in properties, not here.
 *
 * @generated from message operand.v1.YouTubeMetadata
 */
class YouTubeMetadata extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new YouTubeMetadata().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new YouTubeMetadata().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new YouTubeMetadata().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(YouTubeMetadata, a, b);
    }
}
exports.YouTubeMetadata = YouTubeMetadata;
YouTubeMetadata.runtime = protobuf_1.proto3;
YouTubeMetadata.typeName = "operand.v1.YouTubeMetadata";
YouTubeMetadata.fields = protobuf_1.proto3.util.newFieldList(() => []);
/**
 * @generated from message operand.v1.SlackMetadata
 */
class SlackMetadata extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string bot_token = 1;
         */
        this.botToken = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new SlackMetadata().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SlackMetadata().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SlackMetadata().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(SlackMetadata, a, b);
    }
}
exports.SlackMetadata = SlackMetadata;
SlackMetadata.runtime = protobuf_1.proto3;
SlackMetadata.typeName = "operand.v1.SlackMetadata";
SlackMetadata.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "bot_token", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "bot_user_id", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
]);
/**
 * @generated from message operand.v1.GithubRepositoryMetadata
 */
class GithubRepositoryMetadata extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string access_token = 1;
         */
        this.accessToken = "";
        /**
         * @generated from field: string owner = 2;
         */
        this.owner = "";
        /**
         * @generated from field: string name = 3;
         */
        this.name = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new GithubRepositoryMetadata().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new GithubRepositoryMetadata().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new GithubRepositoryMetadata().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(GithubRepositoryMetadata, a, b);
    }
}
exports.GithubRepositoryMetadata = GithubRepositoryMetadata;
GithubRepositoryMetadata.runtime = protobuf_1.proto3;
GithubRepositoryMetadata.typeName = "operand.v1.GithubRepositoryMetadata";
GithubRepositoryMetadata.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "access_token", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "owner", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 3, name: "name", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 4, name: "ref", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
]);
/**
 * @generated from message operand.v1.SourceCodeMetadata
 */
class SourceCodeMetadata extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: operand.v1.ProgrammingLanguage language = 1;
         */
        this.language = ProgrammingLanguage.UNSPECIFIED;
        /**
         * @generated from field: string source_code = 2;
         */
        this.sourceCode = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new SourceCodeMetadata().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SourceCodeMetadata().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SourceCodeMetadata().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(SourceCodeMetadata, a, b);
    }
}
exports.SourceCodeMetadata = SourceCodeMetadata;
SourceCodeMetadata.runtime = protobuf_1.proto3;
SourceCodeMetadata.typeName = "operand.v1.SourceCodeMetadata";
SourceCodeMetadata.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "language", kind: "enum", T: protobuf_1.proto3.getEnumType(ProgrammingLanguage) },
    { no: 2, name: "source_code", kind: "scalar", T: 9 /* ScalarType.STRING */ },
]);
/**
 * @generated from message operand.v1.NotionMetadata
 */
class NotionMetadata extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string token = 1;
         */
        this.token = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new NotionMetadata().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new NotionMetadata().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new NotionMetadata().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(NotionMetadata, a, b);
    }
}
exports.NotionMetadata = NotionMetadata;
NotionMetadata.runtime = protobuf_1.proto3;
NotionMetadata.typeName = "operand.v1.NotionMetadata";
NotionMetadata.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "token", kind: "scalar", T: 9 /* ScalarType.STRING */ },
]);
/**
 * @generated from message operand.v1.DiscordMetadata
 */
class DiscordMetadata extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string guild_id = 1;
         */
        this.guildId = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new DiscordMetadata().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new DiscordMetadata().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new DiscordMetadata().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(DiscordMetadata, a, b);
    }
}
exports.DiscordMetadata = DiscordMetadata;
DiscordMetadata.runtime = protobuf_1.proto3;
DiscordMetadata.typeName = "operand.v1.DiscordMetadata";
DiscordMetadata.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "guild_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
]);
/**
 * @generated from message operand.v1.LinearMetadata
 */
class LinearMetadata extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string access_token = 1;
         */
        this.accessToken = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new LinearMetadata().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new LinearMetadata().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new LinearMetadata().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(LinearMetadata, a, b);
    }
}
exports.LinearMetadata = LinearMetadata;
LinearMetadata.runtime = protobuf_1.proto3;
LinearMetadata.typeName = "operand.v1.LinearMetadata";
LinearMetadata.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "access_token", kind: "scalar", T: 9 /* ScalarType.STRING */ },
]);
/**
 * @generated from message operand.v1.MeetingBotMetadata
 */
class MeetingBotMetadata extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new MeetingBotMetadata().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new MeetingBotMetadata().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new MeetingBotMetadata().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(MeetingBotMetadata, a, b);
    }
}
exports.MeetingBotMetadata = MeetingBotMetadata;
MeetingBotMetadata.runtime = protobuf_1.proto3;
MeetingBotMetadata.typeName = "operand.v1.MeetingBotMetadata";
MeetingBotMetadata.fields = protobuf_1.proto3.util.newFieldList(() => []);
/**
 * @generated from message operand.v1.Properties
 */
class Properties extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: map<string, operand.v1.Property> properties = 1;
         */
        this.properties = {};
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new Properties().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new Properties().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new Properties().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(Properties, a, b);
    }
}
exports.Properties = Properties;
Properties.runtime = protobuf_1.proto3;
Properties.typeName = "operand.v1.Properties";
Properties.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "properties", kind: "map", K: 9 /* ScalarType.STRING */, V: { kind: "message", T: Property } },
]);
/**
 * @generated from message operand.v1.Property
 */
class Property extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: bool indexed = 1;
         */
        this.indexed = false;
        /**
         * @generated from oneof operand.v1.Property.value
         */
        this.value = { case: undefined };
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new Property().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new Property().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new Property().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(Property, a, b);
    }
}
exports.Property = Property;
Property.runtime = protobuf_1.proto3;
Property.typeName = "operand.v1.Property";
Property.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "indexed", kind: "scalar", T: 8 /* ScalarType.BOOL */ },
    { no: 2, name: "text", kind: "scalar", T: 9 /* ScalarType.STRING */, oneof: "value" },
    { no: 3, name: "number", kind: "scalar", T: 1 /* ScalarType.DOUBLE */, oneof: "value" },
    { no: 4, name: "text_array", kind: "message", T: TextArray, oneof: "value" },
    { no: 5, name: "number_array", kind: "message", T: NumberArray, oneof: "value" },
]);
/**
 * @generated from message operand.v1.TextArray
 */
class TextArray extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: repeated string values = 1;
         */
        this.values = [];
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new TextArray().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new TextArray().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new TextArray().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(TextArray, a, b);
    }
}
exports.TextArray = TextArray;
TextArray.runtime = protobuf_1.proto3;
TextArray.typeName = "operand.v1.TextArray";
TextArray.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "values", kind: "scalar", T: 9 /* ScalarType.STRING */, repeated: true },
]);
/**
 * @generated from message operand.v1.NumberArray
 */
class NumberArray extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: repeated double values = 1;
         */
        this.values = [];
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new NumberArray().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new NumberArray().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new NumberArray().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(NumberArray, a, b);
    }
}
exports.NumberArray = NumberArray;
NumberArray.runtime = protobuf_1.proto3;
NumberArray.typeName = "operand.v1.NumberArray";
NumberArray.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "values", kind: "scalar", T: 1 /* ScalarType.DOUBLE */, repeated: true },
]);
/**
 * @generated from message operand.v1.UserProfile
 */
class UserProfile extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string public_id = 1;
         */
        this.publicId = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new UserProfile().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new UserProfile().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new UserProfile().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(UserProfile, a, b);
    }
}
exports.UserProfile = UserProfile;
UserProfile.runtime = protobuf_1.proto3;
UserProfile.typeName = "operand.v1.UserProfile";
UserProfile.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "public_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "email", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 3, name: "handle", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 4, name: "created_at", kind: "message", T: protobuf_1.Timestamp },
    { no: 5, name: "name", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 6, name: "bio", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 7, name: "avatar_url", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
]);
/**
 * @generated from message operand.v1.Author
 */
class Author extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from oneof operand.v1.Author.author
         */
        this.author = { case: undefined };
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new Author().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new Author().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new Author().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(Author, a, b);
    }
}
exports.Author = Author;
Author.runtime = protobuf_1.proto3;
Author.typeName = "operand.v1.Author";
Author.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "profile", kind: "message", T: UserProfile, oneof: "author" },
    { no: 2, name: "email", kind: "scalar", T: 9 /* ScalarType.STRING */, oneof: "author" },
]);
/**
 * @generated from message operand.v1.ObjectPreview
 */
class ObjectPreview extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: repeated operand.v1.Author authors = 5;
         */
        this.authors = [];
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new ObjectPreview().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ObjectPreview().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ObjectPreview().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(ObjectPreview, a, b);
    }
}
exports.ObjectPreview = ObjectPreview;
ObjectPreview.runtime = protobuf_1.proto3;
ObjectPreview.typeName = "operand.v1.ObjectPreview";
ObjectPreview.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "title", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 2, name: "description", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 3, name: "image", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 4, name: "url", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 5, name: "authors", kind: "message", T: Author, repeated: true },
]);
/**
 * @generated from message operand.v1.IndexingProgress
 */
class IndexingProgress extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * 0.0 - 1.0
         *
         * @generated from field: float percent_ready = 1;
         */
        this.percentReady = 0;
        /**
         * @generated from field: map<string, int32> status_counts = 2;
         */
        this.statusCounts = {};
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new IndexingProgress().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new IndexingProgress().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new IndexingProgress().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(IndexingProgress, a, b);
    }
}
exports.IndexingProgress = IndexingProgress;
IndexingProgress.runtime = protobuf_1.proto3;
IndexingProgress.typeName = "operand.v1.IndexingProgress";
IndexingProgress.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "percent_ready", kind: "scalar", T: 2 /* ScalarType.FLOAT */ },
    { no: 2, name: "status_counts", kind: "map", K: 9 /* ScalarType.STRING */, V: { kind: "scalar", T: 5 /* ScalarType.INT32 */ } },
]);
/**
 * @generated from message operand.v1.Object
 */
class Object$ extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string id = 1;
         */
        this.id = "";
        /**
         * @generated from field: operand.v1.ObjectType type = 5;
         */
        this.type = ObjectType.UNSPECIFIED;
        /**
         * @generated from field: operand.v1.ObjectStatus status = 8;
         */
        this.status = ObjectStatus.UNSPECIFIED;
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new Object$().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new Object$().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new Object$().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(Object$, a, b);
    }
}
exports.Object$ = Object$;
Object$.runtime = protobuf_1.proto3;
Object$.typeName = "operand.v1.Object";
Object$.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "created_at", kind: "message", T: protobuf_1.Timestamp },
    { no: 3, name: "updated_at", kind: "message", T: protobuf_1.Timestamp, opt: true },
    { no: 4, name: "parent_id", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 5, name: "type", kind: "enum", T: protobuf_1.proto3.getEnumType(ObjectType) },
    { no: 6, name: "metadata", kind: "message", T: ObjectMetadata, opt: true },
    { no: 7, name: "properties", kind: "message", T: Properties },
    { no: 8, name: "status", kind: "enum", T: protobuf_1.proto3.getEnumType(ObjectStatus) },
    { no: 9, name: "preview", kind: "message", T: ObjectPreview, opt: true },
    { no: 10, name: "indexing_progress", kind: "message", T: IndexingProgress, opt: true },
    { no: 11, name: "error", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
]);
/**
 * @generated from message operand.v1.ObjectOptions
 */
class ObjectOptions extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new ObjectOptions().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ObjectOptions().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ObjectOptions().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(ObjectOptions, a, b);
    }
}
exports.ObjectOptions = ObjectOptions;
ObjectOptions.runtime = protobuf_1.proto3;
ObjectOptions.typeName = "operand.v1.ObjectOptions";
ObjectOptions.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "include_metadata", kind: "scalar", T: 8 /* ScalarType.BOOL */, opt: true },
    { no: 2, name: "include_preview", kind: "scalar", T: 8 /* ScalarType.BOOL */, opt: true },
    { no: 3, name: "include_indexing_progress", kind: "scalar", T: 8 /* ScalarType.BOOL */, opt: true },
]);
/**
 * @generated from message operand.v1.UpsertRequest
 */
class UpsertRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new UpsertRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new UpsertRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new UpsertRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(UpsertRequest, a, b);
    }
}
exports.UpsertRequest = UpsertRequest;
UpsertRequest.runtime = protobuf_1.proto3;
UpsertRequest.typeName = "operand.v1.UpsertRequest";
UpsertRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "existing_id", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 2, name: "parent_id", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 3, name: "type", kind: "enum", T: protobuf_1.proto3.getEnumType(ObjectType), opt: true },
    { no: 4, name: "metadata", kind: "message", T: ObjectMetadata, opt: true },
    { no: 5, name: "properties", kind: "message", T: Properties, opt: true },
    { no: 6, name: "unique_property", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
]);
/**
 * @generated from message operand.v1.UpsertResponse
 */
class UpsertResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new UpsertResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new UpsertResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new UpsertResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(UpsertResponse, a, b);
    }
}
exports.UpsertResponse = UpsertResponse;
UpsertResponse.runtime = protobuf_1.proto3;
UpsertResponse.typeName = "operand.v1.UpsertResponse";
UpsertResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "object", kind: "message", T: Object$ },
]);
/**
 * @generated from message operand.v1.SuggestionsRequest
 */
class SuggestionsRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string query = 1;
         */
        this.query = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new SuggestionsRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SuggestionsRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SuggestionsRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(SuggestionsRequest, a, b);
    }
}
exports.SuggestionsRequest = SuggestionsRequest;
SuggestionsRequest.runtime = protobuf_1.proto3;
SuggestionsRequest.typeName = "operand.v1.SuggestionsRequest";
SuggestionsRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "query", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "type", kind: "enum", T: protobuf_1.proto3.getEnumType(ObjectType), opt: true },
]);
/**
 * @generated from message operand.v1.SuggestionsResponse
 */
class SuggestionsResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: repeated operand.v1.SuggestionsResponse.AnnotatedUpsert upserts = 1;
         */
        this.upserts = [];
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new SuggestionsResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SuggestionsResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SuggestionsResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(SuggestionsResponse, a, b);
    }
}
exports.SuggestionsResponse = SuggestionsResponse;
SuggestionsResponse.runtime = protobuf_1.proto3;
SuggestionsResponse.typeName = "operand.v1.SuggestionsResponse";
SuggestionsResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "upserts", kind: "message", T: SuggestionsResponse_AnnotatedUpsert, repeated: true },
]);
/**
 * @generated from message operand.v1.SuggestionsResponse.AnnotatedUpsert
 */
class SuggestionsResponse_AnnotatedUpsert extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string description = 2;
         */
        this.description = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new SuggestionsResponse_AnnotatedUpsert().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SuggestionsResponse_AnnotatedUpsert().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SuggestionsResponse_AnnotatedUpsert().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(SuggestionsResponse_AnnotatedUpsert, a, b);
    }
}
exports.SuggestionsResponse_AnnotatedUpsert = SuggestionsResponse_AnnotatedUpsert;
SuggestionsResponse_AnnotatedUpsert.runtime = protobuf_1.proto3;
SuggestionsResponse_AnnotatedUpsert.typeName = "operand.v1.SuggestionsResponse.AnnotatedUpsert";
SuggestionsResponse_AnnotatedUpsert.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "upsert", kind: "message", T: UpsertRequest },
    { no: 2, name: "description", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 3, name: "page_estimate", kind: "scalar", T: 5 /* ScalarType.INT32 */, opt: true },
]);
/**
 * @generated from message operand.v1.ListRequest
 */
class ListRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new ListRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ListRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ListRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(ListRequest, a, b);
    }
}
exports.ListRequest = ListRequest;
ListRequest.runtime = protobuf_1.proto3;
ListRequest.typeName = "operand.v1.ListRequest";
ListRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "parent_id", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 2, name: "cursor", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    { no: 3, name: "limit", kind: "scalar", T: 5 /* ScalarType.INT32 */, opt: true },
    { no: 4, name: "options", kind: "message", T: ObjectOptions, opt: true },
]);
/**
 * @generated from message operand.v1.ListResponse
 */
class ListResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: repeated operand.v1.Object objects = 1;
         */
        this.objects = [];
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new ListResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ListResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ListResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(ListResponse, a, b);
    }
}
exports.ListResponse = ListResponse;
ListResponse.runtime = protobuf_1.proto3;
ListResponse.typeName = "operand.v1.ListResponse";
ListResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "objects", kind: "message", T: Object$, repeated: true },
    { no: 2, name: "next_cursor", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
]);
/**
 * @generated from message operand.v1.GetRequest
 */
class GetRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: repeated string ids = 1;
         */
        this.ids = [];
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new GetRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new GetRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new GetRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(GetRequest, a, b);
    }
}
exports.GetRequest = GetRequest;
GetRequest.runtime = protobuf_1.proto3;
GetRequest.typeName = "operand.v1.GetRequest";
GetRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "ids", kind: "scalar", T: 9 /* ScalarType.STRING */, repeated: true },
    { no: 2, name: "options", kind: "message", T: ObjectOptions, opt: true },
]);
/**
 * @generated from message operand.v1.GetResponse
 */
class GetResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: repeated operand.v1.Object objects = 1;
         */
        this.objects = [];
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new GetResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new GetResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new GetResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(GetResponse, a, b);
    }
}
exports.GetResponse = GetResponse;
GetResponse.runtime = protobuf_1.proto3;
GetResponse.typeName = "operand.v1.GetResponse";
GetResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "objects", kind: "message", T: Object$, repeated: true },
]);
/**
 * @generated from message operand.v1.DeleteRequest
 */
class DeleteRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: repeated string ids = 1;
         */
        this.ids = [];
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new DeleteRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new DeleteRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new DeleteRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(DeleteRequest, a, b);
    }
}
exports.DeleteRequest = DeleteRequest;
DeleteRequest.runtime = protobuf_1.proto3;
DeleteRequest.typeName = "operand.v1.DeleteRequest";
DeleteRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "ids", kind: "scalar", T: 9 /* ScalarType.STRING */, repeated: true },
]);
/**
 * @generated from message operand.v1.DeleteResponse
 */
class DeleteResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new DeleteResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new DeleteResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new DeleteResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(DeleteResponse, a, b);
    }
}
exports.DeleteResponse = DeleteResponse;
DeleteResponse.runtime = protobuf_1.proto3;
DeleteResponse.typeName = "operand.v1.DeleteResponse";
DeleteResponse.fields = protobuf_1.proto3.util.newFieldList(() => []);
/**
 * @generated from message operand.v1.CountRequest
 */
class CountRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new CountRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new CountRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new CountRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(CountRequest, a, b);
    }
}
exports.CountRequest = CountRequest;
CountRequest.runtime = protobuf_1.proto3;
CountRequest.typeName = "operand.v1.CountRequest";
CountRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "parent_id", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
]);
/**
 * @generated from message operand.v1.CountResponse
 */
class CountResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: int32 count = 1;
         */
        this.count = 0;
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new CountResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new CountResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new CountResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(CountResponse, a, b);
    }
}
exports.CountResponse = CountResponse;
CountResponse.runtime = protobuf_1.proto3;
CountResponse.typeName = "operand.v1.CountResponse";
CountResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "count", kind: "scalar", T: 5 /* ScalarType.INT32 */ },
]);
/**
 * @generated from message operand.v1.Filter
 */
class Filter extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: repeated operand.v1.Condition conditions = 1;
         */
        this.conditions = [];
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new Filter().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new Filter().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new Filter().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(Filter, a, b);
    }
}
exports.Filter = Filter;
Filter.runtime = protobuf_1.proto3;
Filter.typeName = "operand.v1.Filter";
Filter.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "conditions", kind: "message", T: Condition, repeated: true },
]);
/**
 * @generated from message operand.v1.NotCondition
 */
class NotCondition extends protobuf_1.Message {
    constructor(data) {
        super();
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new NotCondition().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new NotCondition().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new NotCondition().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(NotCondition, a, b);
    }
}
exports.NotCondition = NotCondition;
NotCondition.runtime = protobuf_1.proto3;
NotCondition.typeName = "operand.v1.NotCondition";
NotCondition.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "condition", kind: "message", T: Condition },
]);
/**
 * @generated from message operand.v1.Condition
 */
class Condition extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from oneof operand.v1.Condition.condition
         */
        this.condition = { case: undefined };
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new Condition().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new Condition().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new Condition().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(Condition, a, b);
    }
}
exports.Condition = Condition;
Condition.runtime = protobuf_1.proto3;
Condition.typeName = "operand.v1.Condition";
Condition.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "property", kind: "message", T: KeyedProperty, oneof: "condition" },
    { no: 2, name: "range", kind: "message", T: Range, oneof: "condition" },
    { no: 3, name: "and", kind: "message", T: Filter, oneof: "condition" },
    { no: 4, name: "or", kind: "message", T: Filter, oneof: "condition" },
    { no: 5, name: "not", kind: "message", T: NotCondition, oneof: "condition" },
]);
/**
 * @generated from message operand.v1.KeyedProperty
 */
class KeyedProperty extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string key = 1;
         */
        this.key = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new KeyedProperty().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new KeyedProperty().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new KeyedProperty().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(KeyedProperty, a, b);
    }
}
exports.KeyedProperty = KeyedProperty;
KeyedProperty.runtime = protobuf_1.proto3;
KeyedProperty.typeName = "operand.v1.KeyedProperty";
KeyedProperty.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "key", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "property", kind: "message", T: Property },
]);
/**
 * @generated from message operand.v1.Range
 */
class Range extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string key = 1;
         */
        this.key = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new Range().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new Range().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new Range().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(Range, a, b);
    }
}
exports.Range = Range;
Range.runtime = protobuf_1.proto3;
Range.typeName = "operand.v1.Range";
Range.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "key", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "lt", kind: "scalar", T: 1 /* ScalarType.DOUBLE */, opt: true },
    { no: 3, name: "lte", kind: "scalar", T: 1 /* ScalarType.DOUBLE */, opt: true },
    { no: 4, name: "gt", kind: "scalar", T: 1 /* ScalarType.DOUBLE */, opt: true },
    { no: 5, name: "gte", kind: "scalar", T: 1 /* ScalarType.DOUBLE */, opt: true },
]);
/**
 * @generated from message operand.v1.SearchWithinRequest
 */
class SearchWithinRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string query = 1;
         */
        this.query = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new SearchWithinRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SearchWithinRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SearchWithinRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(SearchWithinRequest, a, b);
    }
}
exports.SearchWithinRequest = SearchWithinRequest;
SearchWithinRequest.runtime = protobuf_1.proto3;
SearchWithinRequest.typeName = "operand.v1.SearchWithinRequest";
SearchWithinRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "query", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "limit", kind: "scalar", T: 5 /* ScalarType.INT32 */, opt: true },
    { no: 3, name: "filter", kind: "message", T: Filter, opt: true },
    { no: 4, name: "object_options", kind: "message", T: ObjectOptions, opt: true },
]);
/**
 * @generated from message operand.v1.SearchWithinResponse
 */
class SearchWithinResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: repeated operand.v1.SearchWithinResponse.Match matches = 1;
         */
        this.matches = [];
        /**
         * @generated from field: map<string, operand.v1.Object> objects = 2;
         */
        this.objects = {};
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new SearchWithinResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SearchWithinResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SearchWithinResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(SearchWithinResponse, a, b);
    }
}
exports.SearchWithinResponse = SearchWithinResponse;
SearchWithinResponse.runtime = protobuf_1.proto3;
SearchWithinResponse.typeName = "operand.v1.SearchWithinResponse";
SearchWithinResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "matches", kind: "message", T: SearchWithinResponse_Match, repeated: true },
    { no: 2, name: "objects", kind: "map", K: 9 /* ScalarType.STRING */, V: { kind: "message", T: Object$ } },
]);
/**
 * @generated from message operand.v1.SearchWithinResponse.Match
 */
class SearchWithinResponse_Match extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string match_id = 1;
         */
        this.matchId = "";
        /**
         * @generated from field: string object_id = 2;
         */
        this.objectId = "";
        /**
         * @generated from field: string content = 3;
         */
        this.content = "";
        /**
         * @generated from field: float score = 4;
         */
        this.score = 0;
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new SearchWithinResponse_Match().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SearchWithinResponse_Match().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SearchWithinResponse_Match().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(SearchWithinResponse_Match, a, b);
    }
}
exports.SearchWithinResponse_Match = SearchWithinResponse_Match;
SearchWithinResponse_Match.runtime = protobuf_1.proto3;
SearchWithinResponse_Match.typeName = "operand.v1.SearchWithinResponse.Match";
SearchWithinResponse_Match.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "match_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "object_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 3, name: "content", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 4, name: "score", kind: "scalar", T: 2 /* ScalarType.FLOAT */ },
    { no: 5, name: "extra", kind: "message", T: Properties, opt: true },
]);
/**
 * Answer is an attempt to answer a question.
 *
 * @generated from message operand.v1.Answer
 */
class Answer extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string answer = 1;
         */
        this.answer = "";
        /**
         * Confidence is a value between 0 and 1, where 1 is the highest confidence.
         * It indicates how confident we are that the answer is factually correct, using
         * the index content as the ground truth.
         *
         * @generated from field: float confidence = 2;
         */
        this.confidence = 0;
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new Answer().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new Answer().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new Answer().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(Answer, a, b);
    }
}
exports.Answer = Answer;
Answer.runtime = protobuf_1.proto3;
Answer.typeName = "operand.v1.Answer";
Answer.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "answer", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "confidence", kind: "scalar", T: 2 /* ScalarType.FLOAT */ },
]);
/**
 * @generated from message operand.v1.AnswerRequest
 */
class AnswerRequest extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: string query = 1;
         */
        this.query = "";
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new AnswerRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new AnswerRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new AnswerRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(AnswerRequest, a, b);
    }
}
exports.AnswerRequest = AnswerRequest;
AnswerRequest.runtime = protobuf_1.proto3;
AnswerRequest.typeName = "operand.v1.AnswerRequest";
AnswerRequest.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "query", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "filter", kind: "message", T: Filter, opt: true },
    { no: 3, name: "object_options", kind: "message", T: ObjectOptions, opt: true },
]);
/**
 * @generated from message operand.v1.AnswerResponse
 */
class AnswerResponse extends protobuf_1.Message {
    constructor(data) {
        super();
        /**
         * @generated from field: repeated operand.v1.Object sources = 2;
         */
        this.sources = [];
        protobuf_1.proto3.util.initPartial(data, this);
    }
    static fromBinary(bytes, options) {
        return new AnswerResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new AnswerResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new AnswerResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return protobuf_1.proto3.util.equals(AnswerResponse, a, b);
    }
}
exports.AnswerResponse = AnswerResponse;
AnswerResponse.runtime = protobuf_1.proto3;
AnswerResponse.typeName = "operand.v1.AnswerResponse";
AnswerResponse.fields = protobuf_1.proto3.util.newFieldList(() => [
    { no: 1, name: "answer", kind: "message", T: Answer, opt: true },
    { no: 2, name: "sources", kind: "message", T: Object$, repeated: true },
]);
