// @bufbuild/connect-web/dist/cjs/grpc-web-transport.js
"use strict";
// Copyright 2021-2022 Buf Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
Object.defineProperty(exports, "__esModule", { value: true });
exports.createGrpcWebTransport = void 0;
const connect_error_js_1 = require("./connect-error.js");
const code_js_1 = require("./code.js");
const http_headers_js_1 = require("./http-headers.js");
const status_pb_js_1 = require("./grpc/status/v1/status_pb.js");
const interceptor_js_1 = require("./interceptor.js");
const envelope_js_1 = require("./envelope.js");
const assert_fetch_api_js_1 = require("./assert-fetch-api.js");
/**
 * Create a Transport for the gRPC-web protocol. The protocol encodes
 * trailers in the response body and makes unary and server-streaming
 * methods available to web browsers. It uses the fetch API to make
 * HTTP requests.
 *
 * Note that this transport does not implement the grpc-web-text format,
 * which applies base64 encoding to the request and response bodies to
 * support reading streaming responses from an XMLHttpRequest.
 */
function createGrpcWebTransport(options) {
    (0, assert_fetch_api_js_1.assertFetchApi)();
    const transportOptions = options;
    return {
        async unary(service, method, signal, timeoutMs, header, message) {
            var _a;
            try {
                return await (0, interceptor_js_1.runUnary)({
                    stream: false,
                    service,
                    method,
                    url: `${options.baseUrl.replace(/\/$/, "")}/${service.typeName}/${method.name}`,
                    init: {
                        method: "POST",
                        credentials: (_a = options.credentials) !== null && _a !== void 0 ? _a : "same-origin",
                        redirect: "error",
                        mode: "cors",
                    },
                    header: createGrpcWebRequestHeaders(header, timeoutMs),
                    message: message instanceof method.I ? message : new method.I(message),
                    signal: signal !== null && signal !== void 0 ? signal : new AbortController().signal,
                }, async (unaryRequest) => {
                    var _a, _b, _c;
                    const response = await fetch(unaryRequest.url, Object.assign(Object.assign({}, unaryRequest.init), { headers: unaryRequest.header, signal: unaryRequest.signal, body: createGrpcWebRequestBody(unaryRequest.message, options.binaryOptions) }));
                    const headError = (_c = (_b = (_a = extractHttpStatusError(response)) !== null && _a !== void 0 ? _a : extractContentTypeError(response.headers)) !== null && _b !== void 0 ? _b : extractDetailsError(response.headers)) !== null && _c !== void 0 ? _c : extractHeadersError(response.headers);
                    if (headError) {
                        throw headError;
                    }
                    if (!response.body) {
                        throw "missing response body";
                    }
                    const reader = (0, envelope_js_1.createEnvelopeReadableStream)(response.body).getReader();
                    const messageOrTrailerResult = await reader.read();
                    if (messageOrTrailerResult.done) {
                        throw "premature eof";
                    }
                    if (messageOrTrailerResult.value.flags === trailerFlag) {
                        // Unary responses require exactly one response message, but in
                        // case of an error, it is perfectly valid to have a response body
                        // that only contains error trailers.
                        parseGrpcWebTrailerAndExtractError(messageOrTrailerResult.value.data);
                        // At this point, we received trailers only, but the trailers did
                        // not have an error status code.
                        throw "unexpected trailer";
                    }
                    const message = method.O.fromBinary(messageOrTrailerResult.value.data, transportOptions.binaryOptions);
                    const trailerResult = await reader.read();
                    if (trailerResult.done) {
                        throw "missing trailer";
                    }
                    if (trailerResult.value.flags !== trailerFlag) {
                        throw "missing trailer";
                    }
                    const trailer = parseGrpcWebTrailerAndExtractError(trailerResult.value.data);
                    const eofResult = await reader.read();
                    if (!eofResult.done) {
                        throw "extraneous data";
                    }
                    return {
                        stream: false,
                        header: response.headers,
                        message,
                        trailer,
                    };
                }, transportOptions.interceptors);
            }
            catch (e) {
                throw (0, connect_error_js_1.connectErrorFromReason)(e, code_js_1.Code.Internal);
            }
        },
        async serverStream(service, method, signal, timeoutMs, header, message) {
            var _a;
            try {
                return await (0, interceptor_js_1.runServerStream)({
                    stream: false,
                    service,
                    method,
                    url: `${options.baseUrl.replace(/\/$/, "")}/${service.typeName}/${method.name}`,
                    init: {
                        method: "POST",
                        credentials: (_a = options.credentials) !== null && _a !== void 0 ? _a : "same-origin",
                        redirect: "error",
                        mode: "cors",
                    },
                    header: createGrpcWebRequestHeaders(header, timeoutMs),
                    message: message instanceof method.I ? message : new method.I(message),
                    signal: signal !== null && signal !== void 0 ? signal : new AbortController().signal,
                }, async (unaryRequest) => {
                    var _a, _b, _c;
                    const response = await fetch(unaryRequest.url, Object.assign(Object.assign({}, unaryRequest.init), { headers: unaryRequest.header, signal: unaryRequest.signal, body: createGrpcWebRequestBody(unaryRequest.message, options.binaryOptions) }));
                    const err = (_c = (_b = (_a = extractHttpStatusError(response)) !== null && _a !== void 0 ? _a : extractContentTypeError(response.headers)) !== null && _b !== void 0 ? _b : extractDetailsError(response.headers)) !== null && _c !== void 0 ? _c : extractHeadersError(response.headers);
                    if (err) {
                        throw err;
                    }
                    if (!response.body) {
                        throw "missing response body";
                    }
                    const reader = (0, envelope_js_1.createEnvelopeReadableStream)(response.body).getReader();
                    let endStreamReceived = false;
                    let messageReceived = false;
                    return {
                        stream: true,
                        service,
                        method,
                        header: response.headers,
                        trailer: new Headers(),
                        async read() {
                            const result = await reader.read();
                            if (result.done) {
                                if (messageReceived && !endStreamReceived) {
                                    throw new connect_error_js_1.ConnectError("missing trailers", code_js_1.Code.Internal);
                                }
                                return {
                                    done: true,
                                    value: undefined,
                                };
                            }
                            if ((result.value.flags & trailerFlag) === trailerFlag) {
                                endStreamReceived = true;
                                const trailer = parseGrpcWebTrailerAndExtractError(result.value.data);
                                trailer.forEach((value, key) => this.trailer.append(key, value));
                                return {
                                    done: true,
                                    value: undefined,
                                };
                            }
                            messageReceived = true;
                            return {
                                done: false,
                                value: method.O.fromBinary(result.value.data, options.binaryOptions),
                            };
                        },
                    };
                }, options.interceptors);
            }
            catch (e) {
                throw (0, connect_error_js_1.connectErrorFromReason)(e, code_js_1.Code.Internal);
            }
        },
    };
}
exports.createGrpcWebTransport = createGrpcWebTransport;
function createGrpcWebRequestBody(message, options) {
    return (0, envelope_js_1.encodeEnvelopes)({
        data: message.toBinary(options),
        flags: messageFlag,
    });
}
const messageFlag = 0b00000000;
const trailerFlag = 0b10000000;
function createGrpcWebRequestHeaders(headers, timeoutMs) {
    const result = new Headers(headers !== null && headers !== void 0 ? headers : {});
    // We provide the most explicit description for our content type.
    // Note that we do not support the grpc-web-text format.
    // https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-WEB.md#protocol-differences-vs-grpc-over-http2
    result.set("Content-Type", "application/grpc-web+proto");
    // Some servers may rely on the request header `X-Grpc-Web` to identify
    // gRPC-web requests. For example the proxy by improbable:
    // https://github.com/improbable-eng/grpc-web/blob/53aaf4cdc0fede7103c1b06f0cfc560c003a5c41/go/grpcweb/wrapper.go#L231
    result.set("X-Grpc-Web", "1");
    // Note that we do not comply with recommended structure for the
    // user-agent string.
    // https://github.com/grpc/grpc/blob/c462bb8d485fc1434ecfae438823ca8d14cf3154/doc/PROTOCOL-HTTP2.md#user-agents
    result.set("X-User-Agent", "@bufbuild/connect-web");
    if (timeoutMs !== undefined) {
        result.set("Grpc-Timeout", `${timeoutMs}m`);
    }
    return result;
}
function extractContentTypeError(header) {
    const type = header.get("Content-Type");
    switch (type === null || type === void 0 ? void 0 : type.toLowerCase()) {
        case "application/grpc-web":
        case "application/grpc-web+proto":
            return undefined;
        case "application/grpc-web-text":
        case "application/grpc-web-text+proto":
            return new connect_error_js_1.ConnectError("grpc-web-text is not supported", code_js_1.Code.Internal);
        case undefined:
        case null:
        default:
            return new connect_error_js_1.ConnectError(`unexpected content type: ${String(type)}`, code_js_1.Code.Internal);
    }
}
function extractHttpStatusError(response) {
    var _a;
    const code = (0, code_js_1.codeFromGrpcWebHttpStatus)(response.status);
    if (!code) {
        return undefined;
    }
    return new connect_error_js_1.ConnectError(decodeURIComponent((_a = response.headers.get("grpc-message")) !== null && _a !== void 0 ? _a : ""), code);
}
function extractHeadersError(header) {
    var _a;
    const value = header.get("grpc-status");
    if (value === null) {
        return undefined;
    }
    const code = parseInt(value);
    if (code === 0) {
        return undefined;
    }
    if (!(code in code_js_1.Code)) {
        return new connect_error_js_1.ConnectError(`invalid grpc-status: ${value}`, code_js_1.Code.Internal, undefined, header);
    }
    return new connect_error_js_1.ConnectError(decodeURIComponent((_a = header.get("grpc-message")) !== null && _a !== void 0 ? _a : ""), code, undefined, header);
}
function extractDetailsError(header) {
    const grpcStatusDetailsBin = header.get("grpc-status-details-bin");
    if (grpcStatusDetailsBin == null) {
        return undefined;
    }
    try {
        const status = (0, http_headers_js_1.decodeBinaryHeader)(grpcStatusDetailsBin, status_pb_js_1.Status);
        // Prefer the protobuf-encoded data to the headers.
        if (status.code == 0) {
            return undefined;
        }
        const error = new connect_error_js_1.ConnectError(status.message, status.code, undefined, header);
        error.details.push(...status.details);
        return error;
    }
    catch (e) {
        const ce = (0, connect_error_js_1.connectErrorFromReason)(e, code_js_1.Code.Internal);
        header.forEach((value, key) => ce.metadata.append(key, value));
        return ce;
    }
}
function parseGrpcWebTrailerAndExtractError(data) {
    var _a;
    const trailer = parseGrpcWebTrailer(data);
    const err = (_a = extractDetailsError(trailer)) !== null && _a !== void 0 ? _a : extractHeadersError(trailer);
    if (err) {
        throw err;
    }
    return trailer;
}
function parseGrpcWebTrailer(data) {
    const headers = new Headers();
    const lines = String.fromCharCode(...data).split("\r\n");
    for (const line of lines) {
        if (line === "") {
            continue;
        }
        const i = line.indexOf(":");
        if (i > 0) {
            const name = line.substring(0, i).trim();
            const value = line.substring(i + 1).trim();
            headers.append(name, value);
        }
    }
    return headers;
}
